---
title: 'Does human land-use lead to biotic homogenisation of forest bird communities in northern USA?'
author: Eric Le Tortorec, Matti Häkkilä, Edmund Zlonis, Gerald Niemi, Mikko Mönkkönen
date: "`r format(Sys.time(), '%d-%m-%Y')`"
output:
  word_document:
    toc: TRUE
    toc_depth: 4
    fig_width: 6
    fig_height: 6
    reference_docx: manuscript_style.docx
---

```{r setup, echo = FALSE, message = FALSE, warning = FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath('/Users/Eric/Dropbox/Eric/Work/JKL/Theses/Matti_Hakkila/paper_4/'))
```

```{r load_packages, echo = FALSE, message = FALSE, warning = FALSE}
library(adephylo)
library(ape)
library(car)
library(cluster)
library(dplyr)
library(ggplot2)
library(gstat)
library(knitr)
library(lattice)
library(maptools)
library(nlme)
library(pander)
library(printr)
library(psych)
library(raster)
library(readxl)
library(reshape2)
library(rgdal)
library(rgeos)
library(spdep)
library(tidyr)
```

```{r set_pander_options, echo = FALSE, message = FALSE, warning = FALSE}
panderOptions('table.split.table', 100)
panderOptions('table.style', 'rmarkdown')
panderOptions('digits', 4)
panderOptions('keep.trailing.zeros', TRUE)
panderOptions('missing', '-')
panderOptions('table.alignment.rownames', 'left')
panderOptions('table.alignment.default', 'left')
panderOptions('table.emphasize.rownames', TRUE)
```

```{r define_functions, echo = FALSE, message = FALSE, warning = FALSE}
figRef <- local({
    tag <- numeric()
    created <- logical()
    used <- logical()
    function(label, caption, prefix = options('figcap.prefix'), 
        sep = options('figcap.sep'), prefix.highlight = options('figcap.prefix.highlight')) {
        i <- which(names(tag) == label)
        if (length(i) == 0) {
            i <- length(tag) + 1
            tag <<- c(tag, i)
            names(tag)[length(tag)] <<- label
            used <<- c(used, FALSE)
            names(used)[length(used)] <<- label
            created <<- c(created, FALSE)
            names(created)[length(created)] <<- label
        }
        if (!missing(caption)) {
            created[label] <<- TRUE
            paste0(prefix.highlight, prefix, ' ', i, sep, prefix.highlight, 
                ' ', caption)
        } else {
            used[label] <<- TRUE
            paste(prefix, tag[label])
        }
    }
})
options(figcap.prefix = 'Figure', figcap.sep = ':', figcap.prefix.highlight = '**')

figRefApp <- local({
    tag <- numeric()
    created <- logical()
    used <- logical()
    function(label, caption, prefix = options('figcap.app.prefix'), 
        sep = options('figcap.app.sep'), prefix.highlight = options('figcap.app.prefix.highlight')) {
        i <- which(names(tag) == label)
        if (length(i) == 0) {
            i <- length(tag) + 1
            tag <<- c(tag, i)
            names(tag)[length(tag)] <<- label
            used <<- c(used, FALSE)
            names(used)[length(used)] <<- label
            created <<- c(created, FALSE)
            names(created)[length(created)] <<- label
        }
        if (!missing(caption)) {
            created[label] <<- TRUE
            paste0(prefix.highlight, prefix, ' ', i, sep, prefix.highlight, 
                ' ', caption)
        } else {
            used[label] <<- TRUE
            paste(prefix, tag[label])
        }
    }
})
options(figcap.app.prefix = 'Figure A', figcap.app.sep = ':', figcap.app.prefix.highlight = '**')

tabRef <- local({
    tag <- numeric()
    created <- logical()
    used <- logical()
    function(label, caption, prefix = options('tabcap.prefix'), 
        sep = options('tabcap.sep'), prefix.highlight = options('tabcap.prefix.highlight')) {
        i <- which(names(tag) == label)
        if (length(i) == 0) {
            i <- length(tag) + 1
            tag <<- c(tag, i)
            names(tag)[length(tag)] <<- label
            used <<- c(used, FALSE)
            names(used)[length(used)] <<- label
            created <<- c(created, FALSE)
            names(created)[length(created)] <<- label
        }
        if (!missing(caption)) {
            created[label] <<- TRUE
            paste0(prefix.highlight, prefix, ' ', i, sep, prefix.highlight, 
                ' ', caption)
        } else {
            used[label] <<- TRUE
            paste(prefix, tag[label])
        }
    }
})
options(tabcap.prefix = 'Table', tabcap.sep = ':', tabcap.prefix.highlight = '**')

tabRefApp <- local({
    tag <- numeric()
    created <- logical()
    used <- logical()
    function(label, caption, prefix = options('tabcap.app.prefix'), 
        sep = options('tabcap.app.sep'), prefix.highlight = options('tabcap.app.prefix.highlight')) {
        i <- which(names(tag) == label)
        if (length(i) == 0) {
            i <- length(tag) + 1
            tag <<- c(tag, i)
            names(tag)[length(tag)] <<- label
            used <<- c(used, FALSE)
            names(used)[length(used)] <<- label
            created <<- c(created, FALSE)
            names(created)[length(created)] <<- label
        }
        if (!missing(caption)) {
            created[label] <<- TRUE
            paste0(prefix.highlight, prefix, ' ', i, sep, prefix.highlight, 
                ' ', caption)
        } else {
            used[label] <<- TRUE
            paste(prefix, tag[label])
        }
    }
})
options(tabcap.app.prefix = 'Table A', tabcap.app.sep = ':', tabcap.app.prefix.highlight = '**')

summary_fun <- function(x,...){
  c(min = min(x, ...), 
    median = median(x, ...), 
    mean = mean(x, ...), 
    max = max(x,...), 
    sd = sd(x, ...), 
    n = length(x))
}

mytheme <- theme(axis.text = element_text(colour = 'black', size = 14), 
  panel.grid.major = element_blank(), 
  panel.grid.minor = element_blank(), 
  panel.background = element_rect(fill = 'NA'), 
  axis.line = element_line(colour = 'black'), 
  axis.title.y = element_text(colour = 'black', size = 16, vjust = 1.2), 
  axis.title.x = element_text(colour = 'black', size = 16, vjust = -0.1), 
  axis.line.x = element_line(color = 'black', size = 0.5), 
  axis.line.y = element_line(color = 'black', size = 0.5))
```

```{r load_data, echo = FALSE, message = FALSE, warning = FALSE, results = 'hide'}
attach('./results/units_merged_fort.rda')
attach('./results/units_merged_beta_subset.rda')
attach('./results/units_merged_beta_subset_fort.rda')
attach('./results/min_points_radius.rda')
attach('./results/bba_forest_point_count.rda')
attach('./results/forest_unit_count.rda')
attach('./results/units_merged_human_impact_subset.rda')
attach('./results/tax_model_summary.rda')
attach('./results/func_model_summary.rda')
attach('./results/phyl_model_summary.rda')
attach('./results/tax_model_vif.rda')
attach('./results/func_model_vif.rda')
attach('./results/phyl_model_vif.rda')
minnesota <- readOGR('./data/minnesota_bird_data/mnbba_homogenization_data_9_22_2015/units_spatial_data/Minnesota_Outline.shp', 'Minnesota_Outline')
```

```{r set_variables, echo = FALSE, message = FALSE, warning = FALSE, results = 'hide'}
expl_names <- c('Intercept', 'Human footprint', 'Forest loss', 'Net primary production', 'Habitat diversity', 'Mean temperature', 'Number forested points')
```
<br>

#Introduction  
* We are facing a biodiversity crisis
* Alpha, beta and gamma diversities are decreasing
* What the loss of beta diversity tells us
  
* What is biotic homogenistation - the anthropogenic blender
* Biotic homogenisation is a pressing worldwide problem 
* How biotic homogenisation can be measured
* Impacts of anthropogenic and natural processes on biotic homogenisation
  
* Taxonomic homogenisation has been studied a fair bit
* Phylogenetic and functional homogenisation have recieved less attention
* Why studying phylogenetic and functional homogenisation is important
  
* The precise mechanisms causing biotic homogenisation are still unclear
* Species introductions and extinctions lead to biotic homogenisation of communities
  
* What we did 
* How this helps us understand the link between human impact and biotic homogenisation better
* Hypotheses:
    1. Increased human footprint and forest loss lead to decreased beta diversity between units
    2. Increased NPP and habitat diversity increase beta diversity between units
    3. Taxonomic beta diversity shows a stronger response to human influence than functional or phylogenetic diversity
<br>

#Methods  
##Study design
The state of Minnesota was split into 617 units, each measuring 12 x 12 miles. Each unit was composed of four townships, forming a square, except for a few specific areas where irregular units constructed from one to four townships were used. A random township in Minnesota was selected, which was used as the northeastern square of a unit. Subsequent units were then formed using the first unit as a reference point until the entire state was covered by units. Within each township three point locations were selected for bird counts, which were conducted as a part of the Minnesota breeding bird atlas (BBA). One BBA point was placed in an area representing the most common land use class in the township, one point was placed in the second most common land use class, and one was placed in a random land use class. 

**A brief description of BBA protocol needed.** A total of 7070 BBA counts were performed between May 2009 and June 2014.  Abundances of birds identified to the species level within `r count_radius`m of the BBA point were used since this was determined to be the maximum distance within which birds could reliably be identified. For those BBA points that had been counted more than once we selected the earlier observation. We used BBA points located at least `r 2 * count_radius`m from each other (twice the distance of the bird count radius) in order to avoid overlap between BBA bird counts. From pairs of BBA points located under `r 2 * count_radius`m from each other we selected the earlier observation, and if both points had the same date one was chosen randomly. For the analyses in this study we used BBA points located in forested land cover classes. The dominant land cover for each BBA point was  determined from classified satellite images (**More information needed**) within `r count_radius`m of the BBA location. For calculating bird community diversities we only used states that had at least `r min_points` BBA points located in a forested land use class (`r bba_forest_point_count` forested BBA points in `r forest_unit_count` units). We determined `r min_points` points to be the minimum number of communities to reliably estimate beta diversity. 
  
##Calculation of beta diversities  
Beta diversity was calcualted as total species diversity per unit minus the mean alpha diversity per unit. This unitless measure reflects the number of individual BBA points with no overlapping species needed to achieve the before- mentioned aplha and gamma diversities.  

* Taxonomic, functional and phylogenetic beta-diversity.
* Calculation of landscape metrics  
    * Human footprint
    * Forest loss
    * Habitat diversity
    * Net primary production
  
##Statistical models  
We chose the explanatory variables so that the variance inflation factors (VIF) were under 3 (Zuur et al. 2007).
All analyses wer performed with `r R.Version()$version.string`.  
* Spatial autocorrelation
<br>

#Results  
##Summary statistics  
```{r summary_stats, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('response_summary', 
  'Summary statistics of forest bird species richness and response variables per unit'))
response_variables <- units_merged_beta_subset@data[c('forest_spp_richness', 
  'tax_beta_div', 'func_beta_div', 'phyl_beta_div')]
summary_table <- as.data.frame(sapply(response_variables, summary_fun))
colnames(summary_table) <- c('Species richness', 'Taxonomic \u03b2', 'Functional \u03b2', 'Phylogenetic \u03b2')
rownames(summary_table) <- c('Min', 'Median', 'Mean', 'Max', 'SD', 'n')
pander::pander(summary_table)
```
  
The range of taxonomic beta diversity values (`r tabRef("response_summary")`) was a lot larger than that of functional and phylogenetic beta diversity, reflecting the more limited set of ecological functions and phylogenetic backgrounds in comparison to the number of species present.  

```{r div_plots, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRef('div_plots', 'Distributions of forest species richness, taxonomic, functional, and phylogenetic beta diversity, and Pearson correlations between diversity measures')}
psych::pairs.panels(units_merged_beta_subset@data[c('forest_spp_richness', 
  'tax_beta_div', 'func_beta_div', 'phyl_beta_div')], hist.col = NULL)
```
  
`r figRef("div_plots")` shows that none of the response variables was strongly correlated with species richness, although taxonomic beta diversity showed a weak positive correlation with unit-wide species richness up to about 70 species, after which the relation levelled off. Species richness was actually very weakly negatively associated with functional and phylogenetic beta diversity. As the figure shows, these beta diversity indicators are invariably low when species richness is high, and vary widely in units with low species richness. In summary, variation in local bird community composition is not strongly linked with regional (unit level) species richness. 
  
All three beta diversity measures were at least moderately positively correlated with each other. The correlation between functional and phylogenetic beta diversity was fairly strong, suggesting that these two aspects of biodiversity are fairly strongly linked with each other in this data set. All three beta diversity measures increased strongly with the number of BBA points per unit, which can be seen in the results of individual response variables.  

##Taxonomic beta diversity  
```{r tax_beta_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('tax_model_summary', 'Summary of spatial error model explaining taxonomic beta diversity'))
tax_model_coef <- as.data.frame(tax_model_summary$Coef)
colnames(tax_model_coef)[4] <- 'p value'
rownames(tax_model_coef) <- expl_names
pander::pander(tax_model_coef)

tax_range <- round(max(units_merged_beta_subset$tax_beta_div) - 
    min(units_merged_beta_subset$tax_beta_div), digits = 3)

min_hdiv <- round(min(units_merged_beta_subset$habitat_div), digits = 2)
max_hdiv <- round(max(units_merged_beta_subset$habitat_div), digits = 2)
hdiv_range <- round(max(units_merged_beta_subset$habitat_div) - 
    min(units_merged_beta_subset$habitat_div), digits = 2)
tax_hdiv_change <- round(tax_model_coef['Habitat diversity', 'Estimate'] * hdiv_range, digits = 2)
tax_hdiv_perc <- round((tax_hdiv_change / tax_range) * 100, digits = 2)
```

Taxonomic beta diversity was not impacted by human impact (human footprint and forest loss), nor by net primary production or mean temperature. The only variable clearly influencing taxonomic beta diversity was habitat diversity, which had a strong positive impact. Along the range of habitat diversity (`r paste(min_hdiv, ' - ', max_hdiv)`), taxonomic beta diversity increased by `r tax_hdiv_change` units, which is `r tax_hdiv_perc`% of the range of taxonomic beta values in our data.
  
##Functional beta diversity  
```{r func_beta_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('func_model_summary', 'Summary of generalised linear model explaining functional beta diversity'))
func_model_coef <- as.data.frame(func_model_summary$coefficients)
colnames(func_model_coef)[4] <- 'p value'
rownames(func_model_coef) <- expl_names
pander::pander(func_model_coef)

func_range <- round(max(units_merged_beta_subset$func_beta_div) - 
    min(units_merged_beta_subset$func_beta_div), digits = 2)

min_hfp <- round(min(units_merged_beta_subset$human_footprint), digits = 2)
max_hfp <- round(max(units_merged_beta_subset$human_footprint), digits = 2)
hfp_range <- round(max(units_merged_beta_subset$human_footprint) - 
    min(units_merged_beta_subset$human_footprint), digits = 2)
func_hfp_change <- round(func_model_coef['Human footprint', 'Estimate'] * hfp_range, digits = 2)
func_hfp_perc <- round((func_hfp_change / func_range) * 100, digits = 2)

min_npp <- round(min(units_merged_beta_subset$net_prim_prod), digits = 2)
max_npp <- round(max(units_merged_beta_subset$net_prim_prod), digits = 2)
npp_range <- round(max(units_merged_beta_subset$net_prim_prod) - 
    min(units_merged_beta_subset$net_prim_prod), digits = 2)
func_npp_change <- round(func_model_coef['Net primary production', 'Estimate'] * npp_range, digits = 2)
func_npp_perc <- round((func_npp_change / func_range) * 100, digits = 2)

min_temp <- round(min(units_merged_beta_subset$mean_temp), digits = 2)
max_temp <- round(max(units_merged_beta_subset$mean_temp), digits = 2)
temp_range <- round(max(units_merged_beta_subset$mean_temp) - 
    min(units_merged_beta_subset$mean_temp), digits = 2)
func_temp_change <- round(func_model_coef['Mean temperature', 'Estimate'] * temp_range, digits = 2)
func_temp_perc <- round((func_temp_change / func_range) * 100, digits = 2)
```
  
On the other hand, functional beta diversity was impacted by human influence. Human footprint had a positive impact on functional beta diversity, even when natural variation caused by habitat diversity, net primary production (npp) and mean annual temperature were taken into account. Along the unitless gradient of human footprint in our data, ranging from `r min_hfp` (very small human footprint) to `r max_hfp` (relatively large human footprint), functional beta diversity increased by `r func_hfp_change`, which is `r func_hfp_perc`% of the range of functional beta values in our data. Forest loss did not appear to influence functional beta diversity. Net primary production had a negative impact on functional beta diversity. Along the range of npp values (`r paste(min_npp, ' - ', max_npp)` kg C / Ha / year) functional beta diversity decreased by `r abs(func_npp_change)` units (`r func_npp_perc`%). Similarly, mean annual temperature had a negative impact, decreasing functional beta diversity by `r abs(func_temp_change)` units (`r func_temp_perc`%) along the temperature gradient ranging from `r paste(min_temp, ' - ', max_temp)`°C.
  
##Phylogenetic beta diversity  
```{r phyl_beta_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('phyl_model_summary', 'Summary of generalised linear model explaining phylogenetic beta diversity'))
phyl_model_coef <- as.data.frame(phyl_model_summary$coefficients)
colnames(phyl_model_coef)[4] <- 'p value'
rownames(phyl_model_coef) <- expl_names
#phyl_model_coef$'Est. Transf.' <- exp(phyl_model_coef$Estimate)
#phyl_model_coef <- dplyr::select(phyl_model_coef, 1, 5, 2, 3, 4)
pander::pander(phyl_model_coef)

phyl_range <- round(max(units_merged_beta_subset$phyl_beta_div) - 
    min(units_merged_beta_subset$phyl_beta_div), digits = 3)

phyl_hfp_change <- round(phyl_model_coef['Human footprint', 'Estimate'] * hfp_range, digits = 2)
phyl_hfp_perc <- round((phyl_hfp_change / func_range) * 100, digits = 2)

min_floss <- round(min(units_merged_beta_subset$forest_loss), digits = 2)
max_floss <- round(max(units_merged_beta_subset$forest_loss), digits = 2)
floss_range <- round(max(units_merged_beta_subset$forest_loss) - 
    min(units_merged_beta_subset$forest_loss), digits = 2)
phyl_floss_change <- round(phyl_model_coef['Forest loss', 'Estimate'] * floss_range, digits = 2)
phyl_floss_perc <- round((phyl_floss_change / phyl_range) * 100, digits = 2)

phyl_npp_perc <- 100 - round(exp(phyl_model_coef['Net primary production', 'Estimate'] * npp_range) * 100, digits = 2)
phyl_npp_change <- round((phyl_npp_perc / 100) * phyl_range, digits = 3)

phyl_temp_perc <- 100 - round(exp(phyl_model_coef['Mean temperature', 'Estimate'] * temp_range) * 100, digits = 2)
phyl_temp_change <- round((phyl_temp_perc / 100) * phyl_range, digits = 3)
```
  
Phylogenetic beta diversity showed a similar pattern to that of functional diversity. Along the gradient of human footprint in our data phylogenetic beta diversity increased by `r phyl_hfp_change`, which is `r phyl_hfp_perc`% of the range of values in our data. Interestingly, forest loss showed the opposite impact to human footprint, increasing phylogenetic beta diversity by `r phyl_floss_perc`% (`r phyl_floss_change` units) along the range of forest loss in our study landscape (`r paste(min_floss, ' - ', max_floss)`%). As with functional beta diversity, npp had a negative impact on phylogenetic beta diversity, although here the biological impacts were a fair bit weaker. Along the range of npp values phylogenetic beta diversity decreased by `r abs(phyl_npp_change)` units (-`r phyl_npp_perc`%). Mean annual temperature also had a negative impact, decreasing phylogenetic beta diversity by `r abs(phyl_temp_change)` units (-`r phyl_temp_perc`%) along the range of temperature values.
<br>

#Discussion  

Summarry of results

Why did human footprint have a positive impact?


Overall, species richness is not a driver of beta-diversity as species rich units (landscapes) do not foster more local variation in bird community composition. Moreover, beta-diversity indices were not positive functions of net primary productivity or ambient temperature. Earlier research has suggested/shown positive relationships. We rather found that beta-diversities were negatively related to NPP and mean temperature. Why would that be?
  
Taxonomic beta-diversity- habitat diversity is the main driver. Theory predicts that different components of species diversity are positively associated with environmental heterogeneity. It is easy to understand why and how HD affect species richness and alpha-diversity (see also Allouche et al. 2012). Our results support earlied findings that habitat heterogeneity also maintains beta-diversity, i.e. differentiation in community composition among sites.
  
Functional beta-diversity was a positive function of human footprint index even after controlling for habitat diversity, productivity etc. Thus, intensified human land-use resulted in increased variation among local communities in their functional composition, i.e. communities in landscapes under strong human land use became functionally dissimilar. I assume a same kind of mechanism is behind this pattern as we suggested in our earlier alpha-diversity paper (Häkkilä et al. 2017): in that paper, we explained why functional diversity, counter to expectation, increases with human land-use intensity in the matrix surrounding the protected areas.
<br>

#Conclusion  
<br>

#Appdendix  
##1. Correlations between human footprint and human influence variables in Minnesota data
```{r hfp_plots, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('hfp_plots', "Distributions of human footprint and other variables describing human influence, and Spearman's rank correlations between variables")}
psych::pairs.panels(units_merged_human_impact_subset, hist.col = NULL, method = 'spearman')
```
  
##2. Correlations between explanatory variables and their distributions
```{r explanatory_plots, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('explanatory_plots', "Distributions of explanatory variables, and Spearman's rank correlations between variables")}
psych::pairs.panels(units_merged_beta_subset@data[c('human_footprint', 
  'forest_loss', 'habitat_div', 'net_prim_prod', 'mean_temp', 'forest_point_freq')], 
  hist.col = NULL, method = 'spearman')
```
  
##2. Correlations between diversity measures
###a) Taxonomic diversity components
```{r tax_plots, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('tax_plots', "Distributions of taxonomic diversity measures, and Pearson correlations between variables")}
psych::pairs.panels(units_merged_beta_subset@data[c(
  'tax_alpha_div', 'tax_beta_div', 'tax_gamma_div')], hist.col = NULL)
```
  
###b) Functional diversity components
```{r func_plots, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('func_plots', "Distributions of functional diversity measures, and Pearson correlations between variables")}
psych::pairs.panels(units_merged_beta_subset@data[c(
  'func_alpha_div', 'func_beta_div', 'func_gamma_div')], hist.col = NULL)
```
  
###c) Phylogenetic diversity components
```{r phyl_plots, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('phyl_plots', "Distributions of phylogenetic diversity measures, and Pearson correlations between variables")}
psych::pairs.panels(units_merged_beta_subset@data[c(
  'phyl_alpha_div', 'phyl_beta_div', 'phyl_gamma_div')], hist.col = NULL)
```
  
##3. Variance inflation factors from models  
```{r vif, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRefApp('vif', 'Variance inflation factors of explanatory variables in models explaining taxonomic, functional, and phylogenetic beta diversity'))
vif_table <- as.data.frame(rbind(t(tax_model_vif), t(func_model_vif), t(phyl_model_vif)))
rownames(vif_table) <- c('Taxonomic', 'Functional', 'Phylogenetic')
colnames(vif_table) <- expl_names[2:7]
pander::pander(vif_table)
```
  
##4. Histogram of number of forest BBA points per unit
```{r for_point_hist, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('for_point_hist', 'Histrogram of the number of forested BBA points per unit')}
ggplot(units_merged_beta_subset@data, aes(x = forest_point_freq)) + 
  geom_histogram(binwidth = 1,  fill = "grey", colour = "black") + 
  xlab('Number of forested points') + 
  ylab('Count') + 
  scale_x_continuous(limits = c(3, 20.1), expand = c(0, 0)) + 
  scale_y_continuous(limits = c(0, 40), expand = c(0, 0)) + 
  mytheme
```
  
##5. Spatial distribution of explanatory variables  
###a) Human footprint  
```{r hfp_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('hfp_plot', 'Map showing the spatial distribution of human footprint values in the state of Minnesota')}
ggplot(data = units_merged_fort, 
  aes_string(x = 'long', y = 'lat', fill = 'human_footprint', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Human \nfootprint') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
  
###b) Forest loss  
```{r for_loss_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('for_loss_plot', 'Map showing the spatial distribution of forest loss values in the state of Minnesota')}
ggplot(data = units_merged_fort, #units_merged_beta_subset_fort
  aes_string(x = 'long', y = 'lat', fill = 'forest_loss', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Forest \nloss') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
  
###c) Habitat diversity  
```{r hab_div_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('hab_div_plot', 'Map showing the spatial distribution of habitat diversity values in the state of Minnesota')}
ggplot(data = units_merged_fort, 
  aes_string(x = 'long', y = 'lat', fill = 'habitat_div', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Habitat \ndiversity') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
  
###d) Net primary production  
```{r npp_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('npp_plot', 'Map showing the spatial distribution of net primary productivity values in the state of Minnesota')}
ggplot(data = units_merged_fort, 
  aes_string(x = 'long', y = 'lat', fill = 'net_prim_prod', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Net primary \nproduction') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
   
###e) Mean temperature  
```{r mean_temp_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('mean_temp_plot', 'Map showing the spatial distribution of mean temperature values in the state of Minnesota')}
ggplot(data = units_merged_fort, 
  aes_string(x = 'long', y = 'lat', fill = 'mean_temp', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Mean \ntemperature') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
  
###f) Number of forested points  
```{r for_points_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('for_points_plot', 'Map showing the spatial distribution of the number of forested BBA points per unit in the state of Minnesota')}
ggplot(data = units_merged_fort, 
  aes_string(x = 'long', y = 'lat', fill = 'forest_point_freq', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Forest point \nfrequency') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
  
##6. List of used R packages and their versions  
```{r list_packages, echo = FALSE, comment = NA}
for (package in sort(.packages())) {
  #print(paste(package, packageVersion(package), sep = ' '))
  message(package, ' ', packageVersion(package))
  #cat(paste(package, ' ', packageVersion(package), '\n'))
}
```