---
output:
  word_document: default
  pdf_document: default
  html_document: default
---

# Title: Does human land-use lead to biotic homogenisation of forest bird communities in northern USA?  
#### Authors: Eric Le Tortorec, Matti Häkkilä, Edmund Zlonis, Gerald Niemi, Mikko Mönkkönen  
#### Date: `r Sys.Date()`  

```{r setup, echo = FALSE, message = FALSE, warning = FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath('/Users/Eric/Dropbox/Eric/Work/JKL/Theses/Matti_Hakkila/paper_4/'))
```
```{r load_packages, echo = FALSE, message = FALSE, warning = FALSE}
library(adephylo)
library(ape)
library(car)
library(cluster)
library(dplyr)
library(ggplot2)
library(gstat)
library(knitr)
library(lattice)
library(maptools)
library(nlme)
library(pander)
library(printr)
library(psych)
library(raster)
library(readxl)
library(reshape2)
library(rgdal)
library(rgeos)
library(spdep)
library(tidyr)
```
```{r set_pander_options, echo = FALSE, message = FALSE, warning = FALSE}
panderOptions('table.split.table', 100)
panderOptions('table.style', 'rmarkdown')
panderOptions('digits', 4)
panderOptions('keep.trailing.zeros', TRUE)
panderOptions('missing', '-')
panderOptions('table.alignment.rownames', 'left')
panderOptions('table.alignment.default', 'left')
panderOptions('table.emphasize.rownames', TRUE)
```
```{r define_functions, echo = FALSE}
figRef <- local({
    tag <- numeric()
    created <- logical()
    used <- logical()
    function(label, caption, prefix = options('figcap.prefix'), 
        sep = options('figcap.sep'), prefix.highlight = options('figcap.prefix.highlight')) {
        i <- which(names(tag) == label)
        if (length(i) == 0) {
            i <- length(tag) + 1
            tag <<- c(tag, i)
            names(tag)[length(tag)] <<- label
            used <<- c(used, FALSE)
            names(used)[length(used)] <<- label
            created <<- c(created, FALSE)
            names(created)[length(created)] <<- label
        }
        if (!missing(caption)) {
            created[label] <<- TRUE
            paste0(prefix.highlight, prefix, ' ', i, sep, prefix.highlight, 
                ' ', caption)
        } else {
            used[label] <<- TRUE
            paste(prefix, tag[label])
        }
    }
})
options(figcap.prefix = 'Figure', figcap.sep = ':', figcap.prefix.highlight = '**')

tabRef <- local({
    tag <- numeric()
    created <- logical()
    used <- logical()
    function(label, caption, prefix = options('tabcap.prefix'), 
        sep = options('tabcap.sep'), prefix.highlight = options('tabcap.prefix.highlight')) {
        i <- which(names(tag) == label)
        if (length(i) == 0) {
            i <- length(tag) + 1
            tag <<- c(tag, i)
            names(tag)[length(tag)] <<- label
            used <<- c(used, FALSE)
            names(used)[length(used)] <<- label
            created <<- c(created, FALSE)
            names(created)[length(created)] <<- label
        }
        if (!missing(caption)) {
            created[label] <<- TRUE
            paste0(prefix.highlight, prefix, ' ', i, sep, prefix.highlight, 
                ' ', caption)
        } else {
            used[label] <<- TRUE
            paste(prefix, tag[label])
        }
    }
})
options(tabcap.prefix = 'Table', tabcap.sep = ':', tabcap.prefix.highlight = '**')

summary_fun <- function(x,...){
  c(min = min(x, ...), 
    median = median(x, ...), 
    mean = mean(x, ...), 
    max = max(x,...), 
    sd = sd(x, ...), 
    n = length(x))
}
```

```{r load_data, echo = FALSE}
attach('./results/units_merged_beta.rda')
attach('./results/tax_model_summary.rda')
attach('./results/func_model_summary.rda')
attach('./results/phyl_model_summary.rda')
attach('./results/min_points_radius.rda')
```

## Introduction  
* We are facing a biodiversity crisis
* Alpha, beta and gamma diversities are decreasing
* What the loss of beta diversity tells us
  
* What is biotic homogenistation - the anthropogenic blender
* Biotic homogenisation is a pressing worldwide problem
* How biotic homogenisation can be measured
* Impacts of anthropogenic and natural processes on biotic homogenisation
  
* Taxonomic homogenisation has been studied a fair bit
* Phylogenetic and functional homogenisation have recieved less attention
* Why studying phylogenetic and functional homogenisation is important
  
* The precise mechanisms causing biotic homogenisation are still unclear
* Species introductions and extinctions lead to biotic homogenisation of communities but 
  
* What we did
* How this helps us understand the link between human impact and biotic homogenisation better
* Hypotheses:
    1. Increased human footprint and forest loss lead to decreased beta diversity between units
    2. Increased npp and habitat diversity increase beta diversity between units
    3. Taxonomic beta diversity shows a stronger response to human influence than functional or phylogenetic diversity
  
----
  
## Methods  
* Outline of Minnesota Breeding Bird Atlas (BBA)
  
* Outline of study design (units)

* Selection of BBA points
The land class classification of BBA points was based on the dominant land class within a `r count_radius`m buffer around the BBA point.The land classes were determined from classified satellite images.Bird observations from each BBA point were restricted to `r count_radius`m, which was determined to be the maximum distance within which birds could reliably be identified. We only selected points that were at least twice this distance from each other (`r 2 * count_radius`m) in order to avoid duplicate counts of bird observations.
In our final analyses we only selected units with at least `r min_points` forested BBA points, which we determined to be the minimum number of communities to reliably estimate beta diversity.
  
* Calculation of beta diversities
  
* Calculation of landscape metrics
    * Human footprint
    * Forest loss
    * Habitat diversity
    * Net primary production
  
* Statistical models
All analyses wer performed with `r R.Version()$version.string`.  
----
  
## Results  
```{r summary_stats, echo = FALSE, results = 'asis', fig.align = 'center'}
response_variables <- units_merged_beta@data[c('forest_spp_richness', 
  'tax_beta_div', 'func_beta_div', 'phyl_beta_div')]
set.caption(tabRef('response_summary', 'Summary statistics of forest bird species richness and response variables per unit'))
pander::pander(sapply(response_variables, summary_fun))
```

$~$

```{r pairplots, echo = FALSE, fig.align = 'center', fig.cap = figRef('pairplots', 'Distributions of, and correlations between response variables and forest species richness in units')}
psych::pairs.panels(units_merged_beta@data[c('forest_spp_richness', 'tax_beta_div', 'func_beta_div', 'phyl_beta_div')])
```

`r figRef("pairplots")` shows that none of the response variables is correlated with forest bird species richness. There are some fairly clear correlations between response variables, with functional and phylogenetic diversity being quite strongly correlated.   

1. Taxonomic beta diversity  
```{r tax_beta_div, echo = FALSE, results = 'asis'}
set.caption(tabRef('tax_model_summary', 'Summary of linear model explaining taxonomic beta diversity'))
pander::pander(tax_model_summary$coefficients)
```

$~$

2. Functional beta diversity  
```{r func_beta_div, echo = FALSE, results = 'asis'}
set.caption(tabRef('func_model_summary', 'Summary of generalised linear model explaining functional beta diversity'))
pander::pander(func_model_summary$coefficients)
```

$~$

3. Phylogenetic beta diversity  
```{r phyl_beta_div, echo = FALSE, results = 'asis'}
pander::pander(phyl_model_summary$Coef)
#print('Under construction')
```
  
----
  
## Discussion  
  
----
  
## Conclusion  
  
----
  
## Appdendix  
1. Spatial distribution of explanatory variables  
![](../figures/explanatory_spatial.png)

2. Variance inflation factors from models  
```{r vif, echo = FALSE}
attach('./results/tax_model_vif.rda')
attach('./results/func_model_vif.rda')
attach('./results/phyl_model_vif.rda')
```

a) Taxonomic beta diversity  
`r tax_model_vif`

b) Functional beta diversity  
`r func_model_vif`

c) Phylogenetic beta diversity
`r phyl_model_vif`

```{r save_plots1, echo = FALSE, message = FALSE, eval = FALSE}
png(filename = './figures/explanatory_spatial.png', height = 1600, width = 800)
multiplot(
  plot_variable_map(units_merged_fort, 'human_footprint'), 
  plot_variable_map(units_merged_fort, 'forest_loss'), 
  plot_variable_map(units_merged_fort, 'net_prim_prod'), 
  plot_variable_map(units_merged_fort, 'habitat_div'), 
  plot_variable_map(units_merged_fort, 'forest_point_freq'), 
  plot_lisa_map(units_merged, human_footprint), 
  plot_lisa_map(units_merged, forest_loss), 
  plot_lisa_map(units_merged, net_prim_prod), 
  plot_lisa_map(units_merged, habitat_div), 
  plot_lisa_map(units_merged, forest_point_freq), 
  cols = 2)
dev.off()
```
  
```{r save_plots2, echo = FALSE, message = FALSE, eval = FALSE}
png(filename = './figures/aaa.png', height = 600, width = 800)
multiplot(
  plot_variable_map(units_merged_fort, 'human_footprint'), 
  plot_variable_map(units_merged_fort, 'forest_loss'), 
  plot_variable_map(units_merged_fort, 'forest_point_freq'), 
  plot_lisa_map(units_merged, human_footprint), 
  plot_lisa_map(units_merged, forest_loss), 
  plot_lisa_map(units_merged, forest_point_freq), 
  plot_variable_map(units_merged_fort, 'net_prim_prod'), 
  plot_variable_map(units_merged_fort, 'habitat_div'), 
  plot.new(), 
  plot_lisa_map(units_merged, net_prim_prod), 
  plot_lisa_map(units_merged, habitat_div), 
  plot.new(), 
  cols = 4)
dev.off()
```
  
```{r, eval = FALSE}
png(filename = './figures/aaa.png', height = 600, width = 800)
par(mfrow=c(3, 4)) 
plot_variable_map(units_merged_fort, 'human_footprint')
plot_variable_map(units_merged_fort, 'forest_loss')
plot_variable_map(units_merged_fort, 'forest_point_freq')
plot_lisa_map(units_merged, human_footprint)
plot_lisa_map(units_merged, forest_loss)
plot_lisa_map(units_merged, forest_point_freq)
plot_variable_map(units_merged_fort, 'net_prim_prod')
plot_variable_map(units_merged_fort, 'habitat_div')
plot.new()
plot_lisa_map(units_merged, net_prim_prod)
plot_lisa_map(units_merged, habitat_div)
plot.new()
dev.off()
```
  
3. List of used R packages and their versions  
```{r list_packages, echo = FALSE, comment = NA, results = 'asis'}
for (package in sort(.packages())) {
  #print(paste(package, packageVersion(package), sep = ' '))
  message(package, ' ', packageVersion(package))
  #cat(paste(package, ' ', packageVersion(package), '\n'))
}
```