---
title: Does human land-use lead to biotic homogenisation of forest bird communities
  in northern USA?
author: "Eric Le Tortorec, Matti Häkkilä, Edmund Zlonis, Gerald Niemi, Mikko Mönkkönen"
date: "`r format(Sys.time(), '%d-%m-%Y')`"
documentclass: "article"
output:
  word_document:
    fig_height: 6
    fig_width: 6
    pandoc_args: -smart
    reference_docx: manuscript_style.docx
    toc: yes
    toc_depth: 4
  html_notebook:
    toc: yes
    toc_depth: 4
csl: ecography.csl
bibliography: manuscript.bib
---
 
```{r setup, echo = FALSE, message = FALSE, warning = FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath('/Users/Eric/Dropbox/Eric/Work/JKL/Theses/Matti_Hakkila/paper_4/'))
```

```{r load_packages, echo = FALSE, message = FALSE, warning = FALSE}
library(adephylo)
library(ape)
library(car)
library(cluster)
library(dplyr)
library(ggplot2)
library(gstat)
library(knitr)
library(lattice)
library(maptools)
library(nlme)
library(pander)
library(printr)
library(psych)
library(raster)
library(readxl)
library(reshape2)
library(rgdal)
library(rgeos)
library(spdep)
library(tidyr)
```

```{r set_pander_options, echo = FALSE, message = FALSE, warning = FALSE}
panderOptions('table.split.table', 150)
panderOptions('table.style', 'rmarkdown')
panderOptions('digits', 3)
panderOptions('keep.trailing.zeros', FALSE)
panderOptions('missing', '-')
panderOptions('table.alignment.rownames', 'left')
panderOptions('table.alignment.default', 'left')
panderOptions('table.emphasize.rownames', TRUE)
#panderOptions('round', 3)
```

```{r define_functions, echo = FALSE, message = FALSE, warning = FALSE}
figRef <- local({
    tag <- numeric()
    created <- logical()
    used <- logical()
    function(label, caption, prefix = options('figcap.prefix'), 
        sep = options('figcap.sep'), prefix.highlight = options('figcap.prefix.highlight')) {
        i <- which(names(tag) == label)
        if (length(i) == 0) {
            i <- length(tag) + 1
            tag <<- c(tag, i)
            names(tag)[length(tag)] <<- label
            used <<- c(used, FALSE)
            names(used)[length(used)] <<- label
            created <<- c(created, FALSE)
            names(created)[length(created)] <<- label
        }
        if (!missing(caption)) {
            created[label] <<- TRUE
            paste0(prefix.highlight, prefix, ' ', i, sep, prefix.highlight, 
                ' ', caption)
        } else {
            used[label] <<- TRUE
            paste(prefix, tag[label])
        }
    }
})
options(figcap.prefix = 'Figure', figcap.sep = ':', figcap.prefix.highlight = '**')

figRefApp <- local({
    tag <- numeric()
    created <- logical()
    used <- logical()
    function(label, caption, prefix = options('figcap.app.prefix'), 
        sep = options('figcap.app.sep'), prefix.highlight = options('figcap.app.prefix.highlight')) {
        i <- which(names(tag) == label)
        if (length(i) == 0) {
            i <- length(tag) + 1
            tag <<- c(tag, i)
            names(tag)[length(tag)] <<- label
            used <<- c(used, FALSE)
            names(used)[length(used)] <<- label
            created <<- c(created, FALSE)
            names(created)[length(created)] <<- label
        }
        if (!missing(caption)) {
            created[label] <<- TRUE
            paste0(prefix.highlight, prefix, ' ', i, sep, prefix.highlight, 
                ' ', caption)
        } else {
            used[label] <<- TRUE
            paste(prefix, tag[label])
        }
    }
})
options(figcap.app.prefix = 'Figure A', figcap.app.sep = ':', figcap.app.prefix.highlight = '**')

tabRef <- local({
    tag <- numeric()
    created <- logical()
    used <- logical()
    function(label, caption, prefix = options('tabcap.prefix'), 
        sep = options('tabcap.sep'), prefix.highlight = options('tabcap.prefix.highlight')) {
        i <- which(names(tag) == label)
        if (length(i) == 0) {
            i <- length(tag) + 1
            tag <<- c(tag, i)
            names(tag)[length(tag)] <<- label
            used <<- c(used, FALSE)
            names(used)[length(used)] <<- label
            created <<- c(created, FALSE)
            names(created)[length(created)] <<- label
        }
        if (!missing(caption)) {
            created[label] <<- TRUE
            paste0(prefix.highlight, prefix, ' ', i, sep, prefix.highlight, 
                ' ', caption)
        } else {
            used[label] <<- TRUE
            paste(prefix, tag[label])
        }
    }
})
options(tabcap.prefix = 'Table', tabcap.sep = ':', tabcap.prefix.highlight = '**')

tabRefApp <- local({
    tag <- numeric()
    created <- logical()
    used <- logical()
    function(label, caption, prefix = options('tabcap.app.prefix'), 
        sep = options('tabcap.app.sep'), prefix.highlight = options('tabcap.app.prefix.highlight')) {
        i <- which(names(tag) == label)
        if (length(i) == 0) {
            i <- length(tag) + 1
            tag <<- c(tag, i)
            names(tag)[length(tag)] <<- label
            used <<- c(used, FALSE)
            names(used)[length(used)] <<- label
            created <<- c(created, FALSE)
            names(created)[length(created)] <<- label
        }
        if (!missing(caption)) {
            created[label] <<- TRUE
            paste0(prefix.highlight, prefix, ' ', i, sep, prefix.highlight, 
                ' ', caption)
        } else {
            used[label] <<- TRUE
            paste(prefix, tag[label])
        }
    }
})
options(tabcap.app.prefix = 'Table A', tabcap.app.sep = ':', tabcap.app.prefix.highlight = '**')

summary_fun <- function(x,...){
  c(min = min(x, ...), 
    median = median(x, ...), 
    mean = mean(x, ...), 
    max = max(x,...), 
    sd = sd(x, ...))
}
```

```{r load_data, echo = FALSE, message = FALSE, warning = FALSE, results = 'hide'}
attach('./results/units_merged_fort.rda')
attach('./results/units_merged_beta_subset.rda')
attach('./results/units_merged_beta_subset_fort.rda')
attach('./results/min_points_radius.rda')
attach('./results/bba_forest_point_count.rda')
attach('./results/forest_unit_count.rda')
attach('./results/units_merged_human_impact_subset.rda')
attach('./results/tax_beta_model_summary.rda')
attach('./results/tax_alpha_model_summary.rda')
attach('./results/tax_gamma_model_summary.rda')
attach('./results/func_beta_model_summary.rda')
attach('./results/func_alpha_model_summary.rda')
attach('./results/func_gamma_model_summary.rda')
attach('./results/phyl_beta_model_summary.rda')
attach('./results/phyl_alpha_model_summary.rda')
attach('./results/phyl_gamma_model_summary.rda')
attach('./results/tax_beta_model_vif.rda')
attach('./results/tax_alpha_model_vif.rda')
attach('./results/tax_gamma_model_vif.rda')
attach('./results/func_beta_model_vif.rda')
attach('./results/func_alpha_model_vif.rda')
attach('./results/func_gamma_model_vif.rda')
attach('./results/phyl_beta_model_vif.rda')
attach('./results/phyl_alpha_model_vif.rda')
attach('./results/phyl_gamma_model_vif.rda')
minnesota <- readOGR('./data/minnesota_bird_data/mnbba_homogenization_data_9_22_2015/units_spatial_data/Minnesota_Outline.shp', 'Minnesota_Outline')
```

```{r calc_ranges, echo = FALSE, message = FALSE, warning = FALSE, results = 'hide'}
min_hfp <- round(min(units_merged_beta_subset$human_footprint), digits = 2)
max_hfp <- round(max(units_merged_beta_subset$human_footprint), digits = 2)
hfp_range <- round(max(units_merged_beta_subset$human_footprint) - 
    min(units_merged_beta_subset$human_footprint), digits = 2)

min_forest <- round(min(units_merged_beta_subset$forest_loss), digits = 2)
max_forest <- round(max(units_merged_beta_subset$forest_loss), digits = 2)
forest_range <- round(max(units_merged_beta_subset$forest_loss) - 
    min(units_merged_beta_subset$forest_loss), digits = 2)

min_npp <- round(min(units_merged_beta_subset$net_prim_prod), digits = 2)
max_npp <- round(max(units_merged_beta_subset$net_prim_prod), digits = 2)
npp_range <- round(max(units_merged_beta_subset$net_prim_prod) - 
    min(units_merged_beta_subset$net_prim_prod), digits = 2)

min_hdiv <- round(min(units_merged_beta_subset$habitat_div), digits = 2)
max_hdiv <- round(max(units_merged_beta_subset$habitat_div), digits = 2)
hdiv_range <- round(max(units_merged_beta_subset$habitat_div) - 
    min(units_merged_beta_subset$habitat_div), digits = 2)

min_temp <- round(min(units_merged_beta_subset$mean_temp), digits = 2)
max_temp <- round(max(units_merged_beta_subset$mean_temp), digits = 2)
temp_range <- round(max(units_merged_beta_subset$mean_temp) - 
    min(units_merged_beta_subset$mean_temp), digits = 2)
```

```{r set_variables, echo = FALSE, message = FALSE, warning = FALSE, results = 'hide'}
expl_names <- c('Intercept', 'Human footprint', 'Forest loss (%)', 'Net primary production (kg C/m^2/Year)', 'Habitat diversity', 'Mean temperature (\u2103 )', 'Number forested points')
```
<br>

# Introduction  
Global studies have shown that human-caused land-use changes and their
associated pressures have led to strong, consistent and accumulating negative
impacts on biodiversity [@HaddadHabitatfragmentationits2015]. Human pressures
such as land expansion for agriculture, timber harvesting, and urbanisation have
significantly modified ecosystems worldwide [@FoleyGlobalConsequencesLand2005].
Of these, forest ecosystems have been especially heavily impacted by human
activities, showing a decrease in area of over 3% during just the last 25 years
[@FAOGlobalforestresources2016] due to forest fires and human activities. There
are signs that the rate of forest loss is decreasing but the proportion of
managed forests is increasing at the expense of natural forest cover
[@FAOGlobalforestresources2016]. An increasing threat to forest worldwide is
harvesting forests for bioenergy (ref). Natural forests harbour high levels of
productivity, biomass, and biodiversity [@FoleyGlobalConsequencesLand2005] and
decreases in their cover have the potential to have significant negative impacts
on biodiversity and ecosystem services.  

There appear to be differing effects of human activities on different scales of
biodiversity [@McGillFifteenformsbiodiversity2015]. Studies have shown that
biodiversity at the local scale (alpha diversity) has not changed significantly
through time, whereas biodiversity at the global scale has decreased. For
example, @VellendGlobalmetaanalysisreveals2013
[-@VellendGlobalmetaanalysisreveals2013] showed that plant species diversity has
not significantly changed through time, and that species increases are as common
as species decreases. Similarly, @DornelasAssemblageTimeSeries2014
[-@DornelasAssemblageTimeSeries2014] showed that time series of local species
richness did not show a systematic loss, although community composition did
change. However, @NewboldGlobaleffectsland2015 [-@NewboldGlobaleffectsland2015]
inferred from models based on empirical data that at a global scale, local
assemblages have lost over 13% of their species richness due to human pressures.
The discrepancy between trends in local and global biodiversity can be explained
by a spatial measure of biodiversity --- beta diversity.  

Beta diversity measures variation in biodiversity in space and, since it cannot
be measured directly, is derived from the total diversity of a region (gamma
diversity) and the local diversities of sampled sites. Beta diversity, in its
additive form, is calculated as gamma diversity divided by the mean alpha
diversity of a region. In this form, beta diversity is a unitless measure that
reflects how many local sites there would be if none of the sites shared common
species. A higher level of beta diversity in a region indicates that the
composition of local sites is different from each other than in a region with a
lower level of beta diversity. Thus, even though local diversity might not show
a decrease, the total diversity of a region could decrease if the individual
sites became more similar. However, in a conservation setting, maximising beta
diversity is not necessarily the goal [@SocolarHowShouldBetaDiversity2016]. For
example, uneven losses in species between sites can lead to decreased mean alpha
diversity, which will increase the beta diversity of a region. On the other
hand, the colonisation of specialists into new sites can increase mean alpha
diversity [@McCuneGainsnativespecies2013], leading to a decrease in the beta
diversity of a region.  

Increasing similarity between local sites (decreased beta diversity), is a
process known as biotic homogenisation [@OldenBiotichomogenizationnew2006].
Biotic homogenisation has been shown to be a global phenomenon, occuring in
almost all taxonomic groups [@BaiserPatternprocessbiotic2012]. Human activities
causing species invasions and extinctions have the potential to cause species
composition in different sites to become more similar
[@BaiserPatternprocessbiotic2012]. Local species extinctions often target
specialists with narrow dietary and habitat preferences, and they are often
replaced by generalists that can withstand fluctuating environments
[@McKinneyBiotichomogenizationfew1999; @ClavelWorldwidedeclinespecialist2011].
These generalists have often been described as species with global ranges but
there is evidence that they are often native species
[@LaSorteTemporalturnovercommon2005]. Since specialists, by definition, possess
specialised functions, their loss in a community decreases functional diversity.
Functional diversity is of special concern for human well-being as it is an
important component of ecosystem functioning, which is the basis of the
ecosystem services that humans depend on [@CardinaleBiodiversitylossits2012].
There is already evidence that the loss of biodiversity is creating an ecosystem
services debt that will be exacerbated by continuing biodiversity loss
[@Isbellbiodiversitydependentecosystemservice2015], with potential decrease in
the resilience of ecosystem functions
[@OliverBiodiversityResilienceEcosystem2015]. In addition, specialised species
have unique phylogenetic histories, the loss of which decreases the phylogenetic
diversity of communities. Since biotic homogenisation, and many other phenomena
related to biodiversity, has been studied mainly from the aspect of the number
of species (taxonomic diversity) we still have a limited understanding of how
different facets of biodiversity respond to human activities.  

We studied how land-use change caused by human actions causes biotic
homogenisation in bird communities of forested habitats in the state of
Minnesota, USA. Our aim was to study if increasing human footprint, which
includes human population, transport infrastructure, and forest loss, causes
sites to become more similar to each other (decreased beta diversity). We also
controlled for factors known to impact biodiversity, including habitat
diversity, net primary productivity, and temperature (refs). We studied this
phenomenon from the taxonomic, functional and phylogenetic viewpoints. We
predicted that increased human footprint and forest loss would lead to decreased
diversity between bird count sites, visible as decreased beta diversity. We also
expected taxonomic, functional, and phylogenetic diversity to show different
sensitivities to human influence.  

<br>

# Methods  
## Study design
Bird point count data were collected between 2009 and 2014 as part of the
Minnesota Breeding Bird Atlas ([MNBBA](https://mnbirdatlas.org/);
@PfannmullerFirstMinnesotaBreeding2017). The goal of the MNBBA was to
systematically sample breeding birds across the state by collecting data in each
township of the state.  Townships measure approximately 93 km^2^ (6 x 6 miles)
and were developed by the Public Land Survey System
([PLSS](https://nationalmap.gov/small_scale/a_plss.html)).  Within each
township, three point locations were selected. One point was placed in an area
representing the most common land use class in the township, one point was
placed in the second most common land use class, and one was placed in a random
land use class. Land use data were derived from the 2001 National Land Cover
Database [@HomerDevelopment2001national2004]. Counts were conducted primarily on
secondary roads; large, paved roads such as state or federal highways were
avoided. Townships in roadless areas such as the Boundary Waters Canoe Area
Wilderness, Voyageurs National Park, and the Red Lake Peatland were sampled from
trails, portages, and water by hiking, biking, boat, or canoe. For the purposes
of this study, the townships in Minnesota were combined into 617 units, each
comprising four townships and measuring roughly 19.3 x 19.3 km (12 x 12 miles).
Each unit formed a square, except for a few specific areas where irregular units
constructed from one to four townships were used. Units were developed by
selecting a random township, which was used as the north-eastern square of the
first unit. Subsequent units were then formed using the first unit as a
reference point until the entire state was covered.  

The primary objective of the MNBBA point counts was to insure equal and
consistent sampling of the bird community across the state of Minnesota. This
included a random sampling approach and standardization of effort in gathering
data within each township. Individuals participating in the gathering of point
counts were required to pass a test of 86 bird species songs or have more than 5
years of field experience in counting birds with point counts in Minnesota.
Those gathering data on point counts were also tested by audiologists to insure
their hearing ability was in normal ranges and they participated in three days
of “standardisation” in point count data gathering with an experienced field
ornithologist.  

During a 10-minute point count, all birds seen or heard were recorded (i.e.,
unlimited distance) but distances from the point were estimated to allow
habitat-specific data to be used (observations within 25, 50, 100 m, and >
100 m of the census point). Point counts were completed from the last week of
May (in southern MN) to the second week of July (northern MN) from 2009 to 2014.
Most counts were completed in June. Point counts were gathered from
approximately 0.5 hour before to four hours after sunrise on days with little
wind (< 15 km/hr) and little or no precipitation. Extensions from sunrise to 6
hours were allowed for counts in western MN due to the high proportion of windy
days and based on the experience of ornithologists in that region. All points
were located and marked with a GPS device, cover type of vegetation visually
estimated, and pictures taken in the two directions perpendicular to the road.  

A total of 7070 MNBBA counts were performed between May 2009 and June 2014.
Abundances of birds identified to the species level within 100m of the MNBBA
point were used since the habitat classification was conducted at the same
spatial scale (see below). For those MNBBA points that had been counted more
than once we selected the earlier observation. We used MNBBA points located at
least 200m from each other (twice the distance of the bird count radius) in
order to avoid overlap between bird counts. From pairs of MNBBA points located
under 200m from each other we selected the earlier observation, and if both
points had the same date one was chosen randomly. For the analyses in this study
we used MNBBA points located in forested land cover classes. In order to ensure
that the dominant habitat being sampled was forest, we first calculated the
proportion of landcover types within 100m of each MNBBA count location using the
‘isectpolyrst’ tool in Geospatial Modelling Environment version 0.7.3.0
[@BeyerGeospatialmodellingenvironment2012]. Landcover classes were derived from
the LANDFIRE dataset [@RollinsoverviewLANDFIREPrototype2006]. One hundred and
fifteen cover types in the LANDFIRE Existing Vegetation Type database were
consolidated and reclassified into 25 classes representing the land-use and
habitat available in Minnesota. Each count location was characterized by the
dominant (highest proportion) landcover type. All counts with any of the
following dominant landcover classes were considered forested; lowland deciduous
forest, northern hardwoods, pine forest, boreal deciduous, oak forest, parkland
deciduous forest, boreal coniferous, lowland coniferous, rural and urban
developed forest, pine-oak barrens and oak savannah. For calculating bird
community diversities, we only used units that had at least 3 MNBBA points
located in a forested land use class (2217 forested MNBBA points in 287 units).
We determined 3 points to be the minimum number of communities to reliably
estimate beta diversity.  
  
## Calculation of beta diversities
Biotic homogenisation, being a temporal process, is usually measured as the
pairwise similarity of a site between two time points.  

Within each unit we calculated taxonomic, functional, and phylogenetic beta
($\beta$) diversities with Rao’s quadratic entropy using R functions developed
by @deBellopartitioningdiversityshowing2010
[-@deBellopartitioningdiversityshowing2010] The Rao index measures dissimilarity
by summing total dissimilarity, and weighting it by species proportions. In
addition, the Rao index makes it easy to incorporate alternative measures of
biodiversity by taking into account distances (e.g. functional or phylogenetic)
between pairs of species [@deBellopartitioningdiversityshowing2010].  

We calculated $\beta_{\text{eqv}}$ as the total diversity of a unit
($\gamma_{\text{eqv}}$) minus the average diversity of MNBBA points
($\alpha_{\text{eqv}}$) within a unit, and calculated $\beta_{\text{prop}}$ as
the proportion of $\gamma_{\text{eqv}}$ explained by $\beta_{\text{eqv}}$. By
taking the number of forested bird count points into account, we were able to
compare the proportion of $\gamma_{\text{eqv}}$ explained by
$\beta_{\text{eqv}}$ between units with differing numbers of points. In
addition, using a proportional measure of $\beta_{\text{eqv}}$ enabled us to
directly compare the proportion explained by different facets of biodiversity
(taxonomic, functional, and phylogenetic).  

Taxonomic diversity measures were calculated using species abundance data
collected from MNBBA bird count locations. We calculated functional diversity by
using numerical data on the diet composition, foraging stratum, activity time
and body size from the dataset produced by @BelmakerRelativerolesecological2015
[-@BelmakerRelativerolesecological2015]. Bird species were limited to those
species found in our data, and functional distances were calculated with the
daisy function in the R package cluster using Euclidean distances. Functional
distances were scaled between 0 and 1, and used as a distance matrix in the
calculation of diversity metrics. For the calculation of phylogenetic diversity
we downloaded a set of 1000 randomly chosen phylogenetic trees from the BirdTree
database [@Jetzglobaldiversitybirds2012], limited to the bird species found in
our data. We used the consense function from the PHYLIP package (v.3.695) to
create a single unrooted concensus tree using the 50% majority rule. This
included all species that appeared in more than 50% of the trees. A phylogenetic
distance matrix for all species in our data was calculated with the distTips
function from the adephylo R package, using the sums of branch lengths. These
distances were then scaled between 0 and 1, and used to calculate phylogenetic
diversities.  

## Calculation of explanatory variables
Human influence was estimated using the human footprint index (HFI) developed by
the Wildlife Conservation Society and the Center for International Earth Science
Information Network [-@WildlifeConservationSociety-WCSLastWildProject2005]. The
HFI uses nine datasets describing four proxies of human influence measured
between 1995 and 2004: population density, land transformation, accessibility,
and electrical power infrastructure [@Sandersonhumanfootprintlast2002]. Each
1km^2^ grid cell in the global dataset was given a value ranging from 0 (low
level) to 10 (high level) for each dataset. The nine values for each grid cell
were summed, and for each biome the cell with the lowest value got a value of 0,
and the cell with the largest value got a value of 100. The HFI data was clipped
to the state of Minnesota, and the average HFI value for each unit was
calculated.  

Forest loss was quantified from the global forest change dataset developed by @HansenHighResolutionGlobalMaps2013 [-@HansenHighResolutionGlobalMaps2013]. The
dataset uses Landsat data to quantify gross forest loss and gain at an annual
frequency between 2000 and 2014. For each unit, we calculated the percent
cover of forest loss between 2000 and 2014. There was no reliable way of
systematically separating forest loss caused by humans from loss caused by
natural events such as forest fires and storms. To limit this potential source
of disturbance, we left out four units that had been significantly impacted by
three large forest fires (Pagami Creek (2011), Cavity Lake (2006) and Ham Lake
(2007) fires). Habitat diversity for each unit was quantified by calculating
the mean alpha diversity of habitat classes within units. Habitat classes were
derived from the LANDFIRE reclassification described above.

We calculated net primary production (NPP)...(to be continued)  
  
## Statistical models  
Linear models and generalised linear models We chose the explanatory variables
so that the variance inflation factors (VIF) were under 3
[@ZuurAnalysingecologicaldata2007]. VIF values can be found in chapter 3 of the
appendix.  

Spatial autocorrelation  

All analyses wer performed with `r R.Version()$version.string`.  

<br>

# Results  
## Summary statistics  
```{r response_summary_stats, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('response_summary', 
  'Summary statistics of forest bird species richness and response variables per unit'))
response_variables <- units_merged_beta_subset@data[c('forest_spp_richness', 
  'tax_beta_div', 'func_beta_div', 'phyl_beta_div')]
response_summary_table <- as.data.frame(sapply(response_variables, summary_fun))
colnames(response_summary_table) <- c('Species richness', 'Taxonomic \u03b2', 'Functional \u03b2', 'Phylogenetic \u03b2')
rownames(response_summary_table) <- c('Min', 'Median', 'Mean', 'Max', 'SD')
pander::pander(response_summary_table)
```
  
The range of taxonomic beta diversity values (`r tabRef("response_summary")`)
was a lot larger than that of functional and phylogenetic beta diversity,
reflecting the more limited set of ecological functions and phylogenetic
backgrounds in comparison to the number of species present.  

```{r explanatory_summary_stats, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('explanatory_summary', 
  'Summary statistics of explanatory variables used in models'))
explanatory_variables <- units_merged_beta_subset@data[c('human_footprint', 
  'forest_loss', 'net_prim_prod', 'habitat_div', 'mean_temp', 'forest_point_freq')]
explanatory_summary_table <- as.data.frame(sapply(explanatory_variables, summary_fun))
colnames(explanatory_summary_table) <- expl_names[2:7]
rownames(explanatory_summary_table) <- c('Min', 'Median', 'Mean', 'Max', 'SD')
pander::pander(explanatory_summary_table)
```

```{r beta_div_plots, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRef('div_plots', 'Distributions of forest species richness, taxonomic, functional, and phylogenetic beta diversity, and Pearson correlations between diversity measures')}
knitr::include_graphics('./figures/beta_div_plots.png')
```
  
`r figRef("div_plots")` shows that none of the response variables was strongly
correlated with species richness. All three beta diversity measures were
positively correlated with each other. The correlation between functional and
phylogenetic beta diversity was fairly strong.  
  
## Taxonomic beta diversity  
```{r tax_beta_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('tax_beta_model_summary', 'Summary of linear model explaining taxonomic beta diversity'))
tax_beta_model_coef <- as.data.frame(tax_beta_model_summary$coefficients)
colnames(tax_beta_model_coef)[4] <- 'p value'
rownames(tax_beta_model_coef) <- expl_names
pander::pander(tax_beta_model_coef)

tax_beta_range <- round(max(units_merged_beta_subset$tax_beta_div) - 
    min(units_merged_beta_subset$tax_beta_div), digits = 3)

tax_beta_hdiv_change <- round(tax_beta_model_coef['Habitat diversity', 'Estimate'] * hdiv_range, digits = 2)
tax_beta_hdiv_perc <- round((tax_beta_hdiv_change / tax_beta_range) * 100, digits = 2)
```

Taxonomic beta diversity was not impacted by human footprint index (HFI), nor
forest loss. Taxonomic beta diversity was positively associated with habitat
diversity. Along the range of habitat diversity values 
(`r tabRef("explanatory_summary")`), taxonomic beta diversity increased by 
`r tax_beta_hdiv_change` units (`r tax_beta_hdiv_perc`%).  
  
## Functional beta diversity  
```{r func_beta_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('func_beta_model_summary', 'Summary of generalised linear model explaining functional beta diversity'))
func_beta_model_coef <- as.data.frame(func_beta_model_summary$coefficients)
colnames(func_beta_model_coef)[4] <- 'p value'
rownames(func_beta_model_coef) <- expl_names
pander::pander(func_beta_model_coef)

func_beta_range <- round(max(units_merged_beta_subset$func_beta_div) - 
    min(units_merged_beta_subset$func_beta_div), digits = 2)

func_beta_hfp_change <- round(func_beta_model_coef['Human footprint', 'Estimate'] * hfp_range, digits = 2)
func_beta_hfp_perc <- round((func_beta_hfp_change / func_beta_range) * 100, digits = 2)

func_beta_npp_change <- round(func_beta_model_coef['Net primary production', 'Estimate'] * npp_range, digits = 2)
func_beta_npp_perc <- round((func_beta_npp_change / func_beta_range) * 100, digits = 2)

func_beta_temp_change <- round(func_beta_model_coef['Mean temperature', 'Estimate'] * temp_range, digits = 2)
func_beta_temp_perc <- round((func_beta_temp_change / func_beta_range) * 100, digits = 2)
```
  
Functional beta diversity had a statistically significant association with HFI.
Along the gradient of HFI in our data (`r tabRef("explanatory_summary")`),
functional beta diversity increased strongly (`r func_beta_hfp_change` units, 
`r func_beta_hfp_perc`%). Forest loss did not appear to influence functional beta
diversity. Interestingly, net primary production (NPP) had a strong negative
impact on functional beta diversity. Along the range of NPP values functional
beta diversity decreased by `r abs(func_beta_npp_change)` units 
(`r func_beta_npp_perc`%). Similarly, mean annual temperature had a negative impact,
decreasing functional beta diversity by `r abs(func_beta_temp_change)` units 
(`r func_beta_temp_perc`%).  
  
## Phylogenetic beta diversity  
```{r phyl_beta_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('phyl_beta_model_summary', 'Summary of spatial error model explaining phylogenetic beta diversity'))
phyl_beta_model_coef <- as.data.frame(phyl_beta_model_summary$Coef)
colnames(phyl_beta_model_coef)[4] <- 'p value'
rownames(phyl_beta_model_coef) <- expl_names
pander::pander(phyl_beta_model_coef)

phyl_beta_range <- round(max(units_merged_beta_subset$phyl_beta_div) - 
    min(units_merged_beta_subset$phyl_beta_div), digits = 3)

phyl_beta_npp_perc <- 100 - round(exp(phyl_beta_model_coef['Net primary production', 'Estimate'] * npp_range) * 100, digits = 2)
phyl_beta_npp_change <- round((phyl_beta_npp_perc / 100) * phyl_beta_range, digits = 3)

phyl_beta_temp_perc <- 100 - round(exp(phyl_beta_model_coef['Mean temperature', 'Estimate'] * temp_range) * 100, digits = 2)
phyl_beta_temp_change <- round((phyl_beta_temp_perc / 100) * phyl_beta_range, digits = 3)
```
  
Similar to taxonomic beta diversity, phylogenetic beta diversity was not
impacted by HFI, nor forest loss. Phylogenetic beta diversity was negatively
associated with NPP and mean temperature, although the biological impacts much
stronger. Along the range of NPP values (`r tabRef("explanatory_summary")`)
phylogenetic beta diversity decreased by `r abs(phyl_beta_npp_change)` units
(-`r phyl_beta_npp_perc`%). Along the range of mean annual temperature values
phylogenetic beta diversity decreased by `r abs(phyl_beta_temp_change)` units
(-`r phyl_beta_temp_perc`%).  

<br>

# Discussion  
Elevated human pressure did not appear to cause biotic homogenisation in our
study site. Increased human footprint index (HFI), which measured the impact of
increased population density, land transformation, accessibility, and electrical
power infrastructure,  did not cause biotic homogenisation in taxonomic,
functional or phylogenetic diversity. Instead, increased human footprint
appeared to cause functional heterogenisation by increasing functional diversity
between sites. Forest loss did not appear to cause biotic homogenisation in any
of the response variables. As such, our results appear to be at odds with
previous studies that have shown homogenising impacts of increased human
pressure. For example, urbanisation has been shown to homogenise environments
and, consequently, biological communities [@McKinneyUrbanizationmajorcause2006;
@SolUrbanisationlossphylogenetic2017], and agricultural expansion and
intensification have been shown to homogenise bacterial
[@RodriguesConversionAmazonrainforest2013] and beetle communities
[@GordonSimplificationcoffeefoliagedwelling2009]. Forest management has also
been shown to cause biotic homogenisation of forest communities, although the
impacts have often been fairly subtle
[@KitchingDistancedrivenspeciesturnover2013;
@IbarraBiotichomogenizationLoss2015;
@MoriBiotichomogenizationdifferentiation2015]. However, it is important to note
that some studies have no biotic homogenisation in response to increased human
pressure [@Lee-CruzImpactLoggingForest2013] or even an increase in between-site
diversity [@CatterallLongtermdynamicsbird2010].  

The result that increased human footprint appeared to increase functional
diversity between sites is contrary to the results of many previous studies
[@McKinneyUrbanizationmajorcause2006]. One potential reason for this could be
that increased human impact raised habitat diversity by adding new land classes,
which in turn added new species into some sites resulting in increased diversity
between sites. Although we took the diversity of land classes within each unit
into account it is possible that increased human impact diversified habitats at
finer scales. Another potential cause of increased beta diversity in response to
increased human impact could be related to the way that beta diversity was
calculated. Beta diversity was calculated as gamma diversity (regional diversity
of a unit) minus mean alpha diversity (mean local diversity of individual MNBBA
sites). Thus, the diversity between sites (beta diversity) could increase in
response to an increase in total diversity or a decrease in mean local
diversity. Human footprint did not have a statistically significant impact on
functional gamma or alpha diversity (see appendix) but the negative estimate for
local diversity was an order of magnitude stronger than that for regional
diversity, which could have caused the positive impact on between-site
diversity.  

Similar to human footprint, forest loss did not appear to cause biotic
homogenisation in any of the responses studied here even though it has been
shown to homogenise biological communities especially in the tropics
[@LoboForestfragmentationdrives2011; @KitchingDistancedrivenspeciesturnover2013;
@IbarraBiotichomogenizationLoss2015]. It is possible that the forests in our
study region, which have experienced heavy logging of natural forests 100-150
years ago [@SchulteHomogenizationnorthernGreat2007], have been homogenised in
terms of species composition and forest structure to such an extent that forest
management activities occuring do not cause further biotic homogenisation.
Indeed, forests in the area still bear a clear signature of previous land use
change [@SchulteHomogenizationnorthernGreat2007]. On the other hand, forest loss
did have a clear impact on local and regional diversity. A loss in forest
coverage did, however, appear to decrease functional and phylogenetic diversity
at the local and regional scales, although the impact on regional functional
diversity was trending. Since forest loss had a negative impact at both local
and regional scales, the method of calculating between-site diversity canceled
out the impact of forest loss on between-site diversity, meaning that no
homogenising impact could be seen. This suggests that even though units lost
local and regional diversity in response to forest loss, MNBBA sites did not
become more alike.  

* Interpret results of biogeographic variables on beta diversities 
* Beta-diversity indices were not positive functions of net primary productivity
or ambient temperature. 
* We found that beta-diversities were negatively related
to NPP and mean temperature. **Why is this?**

-Interpret results from conservation viewpoint Our results suggest that
increased human footprint and forest loss are not a cause of biotic
homogenisation in our study site. Although our study did not have a temporal
dimension to it, the spatial replicates (units) that had clearly differing
levels of human footprint and forest loss should have overcome this. Like
several previous studies have shown, concentrating only on taxonomic diversity
would not have shown any impact of human influence on local, regional or
between-site diversity (refs).  

<br>

# Conclusion  
<br>

# Appendix  
## Taxonomic alpha diversity results
```{r tax_alpha_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('tax_alpha_model_summary', 'Summary of spatial error model explaining taxonomic alpha diversity'))
tax_alpha_model_coef <- as.data.frame(tax_alpha_model_summary$Coef)
colnames(tax_alpha_model_coef)[4] <- 'p value'
rownames(tax_alpha_model_coef) <- expl_names
pander::pander(tax_alpha_model_coef)

tax_alpha_range <- round(max(units_merged_beta_subset$tax_alpha_div) - 
    min(units_merged_beta_subset$tax_alpha_div), digits = 3)
```

## Taxonomic gamma diversity results
```{r tax_gamma_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('tax_gamma_model_summary', 'Summary of spatial error model explaining taxonomic gamma diversity'))
tax_gamma_model_coef <- as.data.frame(tax_gamma_model_summary$Coef)
colnames(tax_gamma_model_coef)[4] <- 'p value'
rownames(tax_gamma_model_coef) <- expl_names
pander::pander(tax_gamma_model_coef)

tax_gamma_range <- round(max(units_merged_beta_subset$tax_gamma_div) - 
    min(units_merged_beta_subset$tax_gamma_div), digits = 3)

tax_gamma_hdiv_change <- round(tax_gamma_model_coef['Habitat diversity', 'Estimate'] * hdiv_range, digits = 2)
tax_gamma_hdiv_perc <- round((tax_gamma_hdiv_change / tax_gamma_range) * 100, digits = 2)
```

## Functional alpha diversity results
```{r func_alpha_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('func_alpha_model_summary', 'Summary of spatial error model explaining functional alpha diversity'))
func_alpha_model_coef <- as.data.frame(func_alpha_model_summary$Coef)
colnames(func_alpha_model_coef)[4] <- 'p value'
rownames(func_alpha_model_coef) <- expl_names
pander::pander(func_alpha_model_coef)

func_alpha_range <- round(max(units_merged_beta_subset$func_alpha_div) - 
    min(units_merged_beta_subset$func_alpha_div), digits = 3)

func_alpha_forest_change <- round(func_alpha_model_coef['Forest loss', 'Estimate'] * forest_range, digits = 2)
func_alpha_forest_perc <- round((func_alpha_forest_change / func_alpha_range) * 100, digits = 2)

func_alpha_npp_change <- round(func_alpha_model_coef['Net primary production', 'Estimate'] * npp_range, digits = 2)
func_alpha_npp_perc <- round((func_alpha_npp_change / func_alpha_range) * 100, digits = 2)

func_alpha_temp_change <- round(func_alpha_model_coef['Mean temperature', 'Estimate'] * temp_range, digits = 2)
func_alpha_temp_perc <- round((func_alpha_temp_change / func_alpha_range) * 100, digits = 2)

func_alpha_hfp_change <- round(func_alpha_model_coef['Human footprint', 'Estimate'] * hfp_range, digits = 2)
func_alpha_hfp_perc <- round((func_alpha_hfp_change / func_alpha_range) * 100, digits = 2)
```

## Functional gamma diversity results
```{r func_gamma_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('func_gamma_model_summary', 'Summary of spatial error model explaining functional gamma diversity'))
func_gamma_model_coef <- as.data.frame(func_gamma_model_summary$Coef)
colnames(func_gamma_model_coef)[4] <- 'p value'
rownames(func_gamma_model_coef) <- expl_names
pander::pander(func_gamma_model_coef)

func_gamma_range <- round(max(units_merged_beta_subset$func_gamma_div) - 
    min(units_merged_beta_subset$func_gamma_div), digits = 3)

func_gamma_forest_change <- round(func_gamma_model_coef['Forest loss', 'Estimate'] * forest_range, digits = 2)
func_gamma_forest_perc <- round((func_gamma_forest_change / func_gamma_range) * 100, digits = 2)

func_gamma_temp_change <- round(func_gamma_model_coef['Mean temperature', 'Estimate'] * temp_range, digits = 2)
func_gamma_temp_perc <- round((func_gamma_temp_change / func_gamma_range) * 100, digits = 2)

func_gamma_hfp_change <- round(func_gamma_model_coef['Human footprint', 'Estimate'] * hfp_range, digits = 4)
func_gamma_hfp_perc <- round((func_gamma_hfp_change / func_alpha_range) * 100, digits = 2)
```

## Phylogenetic alpha diversity results
```{r phyl_alpha_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('phyl_alpha_model_summary', 'Summary of spatial error model explaining phylogenetic alpha diversity'))
phyl_alpha_model_coef <- as.data.frame(phyl_alpha_model_summary$Coef)
colnames(phyl_alpha_model_coef)[4] <- 'p value'
rownames(phyl_alpha_model_coef) <- expl_names
pander::pander(phyl_alpha_model_coef)

phyl_alpha_range <- round(max(units_merged_beta_subset$phyl_alpha_div) - 
    min(units_merged_beta_subset$phyl_alpha_div), digits = 3)

phyl_alpha_hfp_change <- round(phyl_alpha_model_coef['Human footprint', 'Estimate'] * hfp_range, digits = 2)
phyl_alpha_hfp_perc <- round((phyl_alpha_hfp_change / phyl_alpha_range) * 100, digits = 2)

phyl_alpha_forest_change <- round(phyl_alpha_model_coef['Forest loss', 'Estimate'] * forest_range, digits = 2)
phyl_alpha_forest_perc <- round((phyl_alpha_forest_change / phyl_alpha_range) * 100, digits = 2)
```

## Phylogenetic gamma diversity results
```{r phyl_gamma_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('phyl_gamma_model_summary', 'Summary of spatial error model explaining phylogenetic gamma diversity'))
phyl_gamma_model_coef <- as.data.frame(phyl_gamma_model_summary$Coef)
colnames(phyl_gamma_model_coef)[4] <- 'p value'
rownames(phyl_gamma_model_coef) <- expl_names
pander::pander(phyl_gamma_model_coef)

phyl_gamma_range <- round(max(units_merged_beta_subset$phyl_gamma_div) - 
    min(units_merged_beta_subset$phyl_gamma_div), digits = 3)

phyl_gamma_forest_change <- round(phyl_gamma_model_coef['Forest loss', 'Estimate'] * forest_range, digits = 2)
phyl_gamma_forest_perc <- round((phyl_gamma_forest_change / phyl_gamma_range) * 100, digits = 2)
```

## 1. Correlations between human footprint and human influence variables in Minnesota data
```{r hfp_plots, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('hfp_plots', "Distributions of human footprint and other variables describing human influence, and Spearman's rank correlations between variables")}
knitr::include_graphics('../figures/human_impact_plots.png')
```
  
## 2. Correlations between explanatory variables and their distributions
```{r explanatory_plots, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('explanatory_plots', "Distributions of explanatory variables, and Spearman's rank correlations between variables")}
knitr::include_graphics('../figures/explanatory_plots.png')
```
  
## 3. Variance inflation factors from models  
```{r vif, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRefApp('vif', 'Variance inflation factors of explanatory variables in models explaining taxonomic, functional, and phylogenetic beta diversity'))
vif_table <- as.data.frame(rbind(t(tax_beta_model_vif), t(func_beta_model_vif), t(phyl_beta_model_vif)))
rownames(vif_table) <- c('Taxonomic', 'Functional', 'Phylogenetic')
colnames(vif_table) <- expl_names[2:7]
pander::pander(vif_table)
```
  
## 4. Histogram of number of forest MNBBA points per unit
```{r for_point_hist, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('for_point_hist', 'Histrogram of the number of forested MNBBA points per unit')}
knitr::include_graphics('../figures/for_point_hist.png')
```
  
## 5. Spatial distribution of response variables and species richness
### a) Taxonomic beta diversity
```{r tax_div_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.height = 10, fig.width = 10, fig.cap = figRefApp('tax_div_plot', 'Map showing the spatial distribution of taxonomic beta diversity in the state of Minnesota')}
knitr::include_graphics('../figures/tax_beta_map.png')
```
  
### b) Functional beta diversity
```{r func_div_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.height = 2, fig.width = 2, fig.cap = figRefApp('func_div_plot', 'Map showing the spatial distribution of functional beta diversity in the state of Minnesota')}
knitr::include_graphics('../figures/func_beta_map.png')
```
  
### c) Phylogenetic beta diversity
```{r phyl_div_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('phyl_div_plot', 'Map showing the spatial distribution of phylogenetic beta diversity in the state of Minnesota')}
knitr::include_graphics('../figures/phyl_beta_map.png')
```
  
### d) Forest species richness
```{r spp_richness_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('spp_richness_plot', 'Map showing the spatial distribution of species richness in the state of Minnesota')}
knitr::include_graphics('../figures/spp_rich_map.png')
```
  
## 6. Spatial distribution of explanatory variables  
### a) Human footprint  
```{r hfp_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('hfp_plot', 'Map showing the spatial distribution of human footprint values in the state of Minnesota')}
knitr::include_graphics('../figures/hfp_map.png')
```
  
### b) Forest loss  
```{r for_loss_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('for_loss_plot', 'Map showing the spatial distribution of forest loss values in the state of Minnesota')}
knitr::include_graphics('../figures/f_loss_map.png')
```
  
### c) Habitat diversity  
```{r hab_div_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('hab_div_plot', 'Map showing the spatial distribution of habitat diversity values in the state of Minnesota')}
knitr::include_graphics('../figures/hab_div_map.png')
```
  
### d) Net primary production  
```{r npp_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('npp_plot', 'Map showing the spatial distribution of net primary productivity values in the state of Minnesota')}
knitr::include_graphics('../figures/npp_map.png')
```
   
### e) Mean temperature  
```{r mean_temp_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('mean_temp_plot', 'Map showing the spatial distribution of mean temperature values in the state of Minnesota')}
knitr::include_graphics('../figures/mean_temp_map.png')
```
  
## 7. List of used R packages and their versions  
```{r list_packages, echo = FALSE, comment = NA}
for (package in sort(.packages())) {
  #print(paste(package, packageVersion(package), sep = ' '))
  message(package, ' ', packageVersion(package))
  #cat(paste(package, ' ', packageVersion(package), '\n'))
}
```

# References