---
title: 'Does human land-use lead to biotic homogenisation of forest bird communities in northern USA?'
author: Eric Le Tortorec, Matti Häkkilä, Edmund Zlonis, Gerald Niemi, Mikko Mönkkönen
date: "`r format(Sys.time(), '%d-%m-%Y')`"
output:
  word_document:
    toc: TRUE
    toc_depth: 4
    fig_width: 6
    fig_height: 6
    reference_docx: manuscript_style.docx
---

```{r setup, echo = FALSE, message = FALSE, warning = FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath('/Users/Eric/Dropbox/Eric/Work/JKL/Theses/Matti_Hakkila/paper_4/'))
```

```{r load_packages, echo = FALSE, message = FALSE, warning = FALSE}
library(adephylo)
library(ape)
library(car)
library(cluster)
library(dplyr)
library(ggplot2)
library(gstat)
library(knitr)
library(lattice)
library(maptools)
library(nlme)
library(pander)
library(printr)
library(psych)
library(raster)
library(readxl)
library(reshape2)
library(rgdal)
library(rgeos)
library(spdep)
library(tidyr)
```

```{r set_pander_options, echo = FALSE, message = FALSE, warning = FALSE}
panderOptions('table.split.table', 150)
panderOptions('table.style', 'rmarkdown')
panderOptions('digits', 3)
panderOptions('keep.trailing.zeros', FALSE)
panderOptions('missing', '-')
panderOptions('table.alignment.rownames', 'left')
panderOptions('table.alignment.default', 'left')
panderOptions('table.emphasize.rownames', TRUE)
#panderOptions('round', 3)
```

```{r define_functions, echo = FALSE, message = FALSE, warning = FALSE}
figRef <- local({
    tag <- numeric()
    created <- logical()
    used <- logical()
    function(label, caption, prefix = options('figcap.prefix'), 
        sep = options('figcap.sep'), prefix.highlight = options('figcap.prefix.highlight')) {
        i <- which(names(tag) == label)
        if (length(i) == 0) {
            i <- length(tag) + 1
            tag <<- c(tag, i)
            names(tag)[length(tag)] <<- label
            used <<- c(used, FALSE)
            names(used)[length(used)] <<- label
            created <<- c(created, FALSE)
            names(created)[length(created)] <<- label
        }
        if (!missing(caption)) {
            created[label] <<- TRUE
            paste0(prefix.highlight, prefix, ' ', i, sep, prefix.highlight, 
                ' ', caption)
        } else {
            used[label] <<- TRUE
            paste(prefix, tag[label])
        }
    }
})
options(figcap.prefix = 'Figure', figcap.sep = ':', figcap.prefix.highlight = '**')

figRefApp <- local({
    tag <- numeric()
    created <- logical()
    used <- logical()
    function(label, caption, prefix = options('figcap.app.prefix'), 
        sep = options('figcap.app.sep'), prefix.highlight = options('figcap.app.prefix.highlight')) {
        i <- which(names(tag) == label)
        if (length(i) == 0) {
            i <- length(tag) + 1
            tag <<- c(tag, i)
            names(tag)[length(tag)] <<- label
            used <<- c(used, FALSE)
            names(used)[length(used)] <<- label
            created <<- c(created, FALSE)
            names(created)[length(created)] <<- label
        }
        if (!missing(caption)) {
            created[label] <<- TRUE
            paste0(prefix.highlight, prefix, ' ', i, sep, prefix.highlight, 
                ' ', caption)
        } else {
            used[label] <<- TRUE
            paste(prefix, tag[label])
        }
    }
})
options(figcap.app.prefix = 'Figure A', figcap.app.sep = ':', figcap.app.prefix.highlight = '**')

tabRef <- local({
    tag <- numeric()
    created <- logical()
    used <- logical()
    function(label, caption, prefix = options('tabcap.prefix'), 
        sep = options('tabcap.sep'), prefix.highlight = options('tabcap.prefix.highlight')) {
        i <- which(names(tag) == label)
        if (length(i) == 0) {
            i <- length(tag) + 1
            tag <<- c(tag, i)
            names(tag)[length(tag)] <<- label
            used <<- c(used, FALSE)
            names(used)[length(used)] <<- label
            created <<- c(created, FALSE)
            names(created)[length(created)] <<- label
        }
        if (!missing(caption)) {
            created[label] <<- TRUE
            paste0(prefix.highlight, prefix, ' ', i, sep, prefix.highlight, 
                ' ', caption)
        } else {
            used[label] <<- TRUE
            paste(prefix, tag[label])
        }
    }
})
options(tabcap.prefix = 'Table', tabcap.sep = ':', tabcap.prefix.highlight = '**')

tabRefApp <- local({
    tag <- numeric()
    created <- logical()
    used <- logical()
    function(label, caption, prefix = options('tabcap.app.prefix'), 
        sep = options('tabcap.app.sep'), prefix.highlight = options('tabcap.app.prefix.highlight')) {
        i <- which(names(tag) == label)
        if (length(i) == 0) {
            i <- length(tag) + 1
            tag <<- c(tag, i)
            names(tag)[length(tag)] <<- label
            used <<- c(used, FALSE)
            names(used)[length(used)] <<- label
            created <<- c(created, FALSE)
            names(created)[length(created)] <<- label
        }
        if (!missing(caption)) {
            created[label] <<- TRUE
            paste0(prefix.highlight, prefix, ' ', i, sep, prefix.highlight, 
                ' ', caption)
        } else {
            used[label] <<- TRUE
            paste(prefix, tag[label])
        }
    }
})
options(tabcap.app.prefix = 'Table A', tabcap.app.sep = ':', tabcap.app.prefix.highlight = '**')

summary_fun <- function(x,...){
  c(min = min(x, ...), 
    median = median(x, ...), 
    mean = mean(x, ...), 
    max = max(x,...), 
    sd = sd(x, ...))
}

mytheme <- theme(axis.text = element_text(colour = 'black', size = 14), 
  panel.grid.major = element_blank(), 
  panel.grid.minor = element_blank(), 
  panel.background = element_rect(fill = 'NA'), 
  axis.line = element_line(colour = 'black'), 
  axis.title.y = element_text(colour = 'black', size = 16, vjust = 1.2), 
  axis.title.x = element_text(colour = 'black', size = 16, vjust = -0.1), 
  axis.line.x = element_line(color = 'black', size = 0.5), 
  axis.line.y = element_line(color = 'black', size = 0.5))
```

```{r load_data, echo = FALSE, message = FALSE, warning = FALSE, results = 'hide'}
attach('./results/units_merged_fort.rda')
attach('./results/units_merged_beta_subset.rda')
attach('./results/units_merged_beta_subset_fort.rda')
attach('./results/min_points_radius.rda')
attach('./results/bba_forest_point_count.rda')
attach('./results/forest_unit_count.rda')
attach('./results/units_merged_human_impact_subset.rda')
attach('./results/tax_beta_model_summary.rda')
attach('./results/tax_alpha_model_summary.rda')
attach('./results/tax_gamma_model_summary.rda')
attach('./results/func_beta_model_summary.rda')
attach('./results/func_alpha_model_summary.rda')
attach('./results/func_gamma_model_summary.rda')
attach('./results/phyl_beta_model_summary.rda')
attach('./results/phyl_alpha_model_summary.rda')
attach('./results/phyl_gamma_model_summary.rda')
attach('./results/tax_beta_model_vif.rda')
attach('./results/tax_alpha_model_vif.rda')
attach('./results/tax_gamma_model_vif.rda')
attach('./results/func_beta_model_vif.rda')
attach('./results/func_alpha_model_vif.rda')
attach('./results/func_gamma_model_vif.rda')
attach('./results/phyl_beta_model_vif.rda')
attach('./results/phyl_alpha_model_vif.rda')
attach('./results/phyl_gamma_model_vif.rda')
minnesota <- readOGR('./data/minnesota_bird_data/mnbba_homogenization_data_9_22_2015/units_spatial_data/Minnesota_Outline.shp', 'Minnesota_Outline')
```

```{r calc_ranges, echo = FALSE, message = FALSE, warning = FALSE, results = 'hide'}
min_hfp <- round(min(units_merged_beta_subset$human_footprint), digits = 2)
max_hfp <- round(max(units_merged_beta_subset$human_footprint), digits = 2)
hfp_range <- round(max(units_merged_beta_subset$human_footprint) - 
    min(units_merged_beta_subset$human_footprint), digits = 2)

min_forest <- round(min(units_merged_beta_subset$forest_loss), digits = 2)
max_forest <- round(max(units_merged_beta_subset$forest_loss), digits = 2)
forest_range <- round(max(units_merged_beta_subset$forest_loss) - 
    min(units_merged_beta_subset$forest_loss), digits = 2)

min_npp <- round(min(units_merged_beta_subset$net_prim_prod), digits = 2)
max_npp <- round(max(units_merged_beta_subset$net_prim_prod), digits = 2)
npp_range <- round(max(units_merged_beta_subset$net_prim_prod) - 
    min(units_merged_beta_subset$net_prim_prod), digits = 2)

min_hdiv <- round(min(units_merged_beta_subset$habitat_div), digits = 2)
max_hdiv <- round(max(units_merged_beta_subset$habitat_div), digits = 2)
hdiv_range <- round(max(units_merged_beta_subset$habitat_div) - 
    min(units_merged_beta_subset$habitat_div), digits = 2)

min_temp <- round(min(units_merged_beta_subset$mean_temp), digits = 2)
max_temp <- round(max(units_merged_beta_subset$mean_temp), digits = 2)
temp_range <- round(max(units_merged_beta_subset$mean_temp) - 
    min(units_merged_beta_subset$mean_temp), digits = 2)
```

```{r set_variables, echo = FALSE, message = FALSE, warning = FALSE, results = 'hide'}
expl_names <- c('Intercept', 'Human footprint', 'Forest loss', 'Net primary production', 'Habitat diversity', 'Mean temperature', 'Number forested points')
```
<br>

#Introduction  
* Discuss beta's relation to alpha and gamma!

* We are facing a biodiversity crisis
* Alpha, beta and gamma diversities are decreasing
* What the loss of beta diversity tells us
  
* What is biotic homogenistation - the anthropogenic blender
* Biotic homogenisation is a pressing worldwide problem 
* How biotic homogenisation can be measured
* Impacts of anthropogenic and natural processes on biotic homogenisation
  
* Taxonomic homogenisation has been studied a fair bit
* Phylogenetic and functional homogenisation have recieved less attention
* Why studying phylogenetic and functional homogenisation is important
  
* The precise mechanisms causing biotic homogenisation are still unclear
* Species introductions and extinctions lead to biotic homogenisation of communities
  
* What we did 
* How this helps us understand the link between human impact and biotic homogenisation better
* Hypotheses:
    1. Increased human footprint and forest loss lead to decreased beta diversity between units
    2. Increased NPP and habitat diversity increase beta diversity between units
    3. Taxonomic beta diversity shows a stronger response to human influence than functional or phylogenetic diversity
<br>

#Methods  
##Study design
The state of Minnesota was split into 617 units, each measuring 12 x 12 miles. Each unit was composed of four townships, forming a square, except for a few specific areas where irregular units constructed from one to four townships were used. A random township in Minnesota was selected, which was used as the northeastern square of a unit. Subsequent units were then formed using the first unit as a reference point until the entire state was covered by units. Within each township three point locations were selected for bird counts, which were conducted as a part of the Minnesota breeding bird atlas (BBA). One BBA point was placed in an area representing the most common land use class in the township, one point was placed in the second most common land use class, and one was placed in a random land use class. 

**A brief description of BBA protocol.** Bird counts were performed between May 2009 and June 2014.

A total of 7070 BBA counts were performed between May 2009 and June 2014.  Abundances of birds identified to the species level within `r count_radius`m of the BBA point were used since this was determined to be the maximum distance within which birds could reliably be identified. For those BBA points that had been counted more than once we selected the earlier observation. We used BBA points located at least `r 2 * count_radius`m from each other (twice the distance of the bird count radius) in order to avoid overlap between BBA bird counts. From pairs of BBA points located under `r 2 * count_radius`m from each other we selected the earlier observation, and if both points had the same date one was chosen randomly. For the analyses in this study we used BBA points located in forested land cover classes. The dominant land cover for each BBA point was  determined from classified satellite images (**Ed, can you add some more information about this?**) within `r count_radius`m of the BBA location. For calculating bird community diversities we only used units that had at least `r min_points` BBA points located in a forested land use class (`r bba_forest_point_count` forested BBA points in `r forest_unit_count` units). We determined `r min_points` points to be the minimum number of communities to reliably estimate beta diversity. 
  
##Calculation of beta diversities
Within each unit we calculated taxonomic, functional, and phylogenetic beta ($\beta$) diversities with Rao’s quadratic entropy using R functions developed by De Bello et al. The Rao index measures dissimilarity by summing total dissimilarity, and weighting it by species proportions. In addition, the Rao index makes it easy to incorporate alternative measures of biodiversity by taking into account distances (e.g. functional or phylogenetic) between pairs of species (de Bello et al. 2010). 

We calculated $\beta_{\text{eqv}}$ as the total diversity of a unit ($\gamma_{\text{eqv}}$) minus the average diversity of BBA points ($\alpha_{\text{eqv}}$) within a unit, and calculated $\beta_{\text{prop}}$ as the proportion of $\gamma_{\text{eqv}}$ explained by $\beta_{\text{eqv}}$. By taking the number of forested bird count points into account int he models we were able to compare the proportion of $\gamma_{\text{eqv}}$ explained by $\beta_{\text{eqv}}$ between units with differing numbers of points. In addition, using a proportional measure of $\beta_{\text{eqv}}$ enabled us to directly compare the proportion explained by different facets of biodiversity (taxonomic, functional, and phylogenetic).

Taxonomic diversity measures were calculated using species abundance data collected from BBA bird count locations. We calculated functional diversity by using numerical data on the diet composition, foraging stratum, activity time and body size from the dataset produced by Belmaker et al. (2014). Bird species were limited to those species found in our data, and functional distances were calculated with the daisy function in the R package cluster using euclidean distances. Functional distances were scaled between 0 and 1, and used as a distance matrix in the calculation of diversity metrics. For the calculation of phylogenetic diversity we downloaded a set of 1000 randomly chosen phylogenetic trees from the BirdTree database (Jetz et al. 2012), limited to the bird species found in our data. We used the consense function from the PHYLIP package (v.3.695) to create a single unrooted concensus tree using the 50% majority rule. This included all species that appeared in more than 50% of the trees. A phylogenetic distance matrix for all species in our data was calculated with the distTips function from the adephylo R package, using the sums of branch lengths. These distances were then scaled between 0 and 1, and used to calculate phylogenetic diversities.

##Calculation of explanatory variables

Human influence was estimated using the human footprint index (HFI) developed by the Wildlife Conservation Society and the Center For International Earth Science Information Network (2005). The HFI uses nine datasets describing four proxies of human influence measured between 1995 and 2004: population density, land transformation, accessibility, and electrical power infrastructure (Sanderson et al. 2002). Each 1km^2 grid cell in the global dataset was given a value ranging from 0 (low level) to 10 (high level) for each dataset. The nine values for each grid cell were summed, and for each biome the cell with the lowest value got a value of 0, and the cell with the largest value got a value of 100. The hfi data was clipped to the state of Minnesota, and the average hfi value for each unit was calculated.

Forest loss was quantified from the the global forest change dataset developed by Hansen et al. (2013). The dataset uses Landsat data to quantify gross forest loss and gain at an annual frequency between 2000 and 2014. For each unit we calculated the percent cover of forest loss between 2000 and 2014. Even though the time span of the forest loss data began nine years before the first BBA observations we assumed that areas impacted by forest loss before the bird counts would have still been non-habitat for the vast majority of the forest birds in our data (**is this a safe assumption?**). There was no reliable way of systematically separating forest loss caused by humans from loss caused by natural events such as forest fires and storms. In order to limit this potential source of confusion we left out four units that had been significantly impacted by two large forest fires (Pagami Creek and Ham Lake fires).

Habitat diversity for each unit was quantified by calculating the mean alpha diversity of habitat classes within units. Habitat classes were taken from... **Ed, could you fill in some basic information about how the landfire3 classification was done?**

We calculated net primary production (NPP)...(to be continued)
  
##Statistical models  
Linear models and generalised linear models
We chose the explanatory variables so that the variance inflation factors (VIF) were under 3 (Zuur et al. 2007). VIF values can be found in chapter 3 of the appendix.

Spatial autocorrelation

All analyses wer performed with `r R.Version()$version.string`.
<br>

#Results  
##Summary statistics  
```{r response_summary_stats, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('response_summary', 
  'Summary statistics of forest bird species richness and response variables per unit'))
response_variables <- units_merged_beta_subset@data[c('forest_spp_richness', 
  'tax_beta_div', 'func_beta_div', 'phyl_beta_div')]
response_summary_table <- as.data.frame(sapply(response_variables, summary_fun))
colnames(response_summary_table) <- c('Species richness', 'Taxonomic \u03b2', 'Functional \u03b2', 'Phylogenetic \u03b2')
rownames(response_summary_table) <- c('Min', 'Median', 'Mean', 'Max', 'SD')
pander::pander(response_summary_table)
```
  
The range of taxonomic beta diversity values (`r tabRef("response_summary")`) was a lot larger than that of functional and phylogenetic beta diversity, reflecting the more limited set of ecological functions and phylogenetic backgrounds in comparison to the number of species present.  

```{r explanatory_summary_stats, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('explanatory_summary', 
  'Summary statistics of explanatory variables used in models'))
explanatory_variables <- units_merged_beta_subset@data[c('human_footprint', 
  'forest_loss', 'net_prim_prod', 'habitat_div', 'mean_temp', 'forest_point_freq')]
explanatory_summary_table <- as.data.frame(sapply(explanatory_variables, summary_fun))
colnames(explanatory_summary_table) <- expl_names[2:7]
rownames(explanatory_summary_table) <- c('Min', 'Median', 'Mean', 'Max', 'SD')
pander::pander(explanatory_summary_table)
```

```{r beta_div_plots, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRef('div_plots', 'Distributions of forest species richness, taxonomic, functional, and phylogenetic beta diversity, and Pearson correlations between diversity measures')}
psych::pairs.panels(units_merged_beta_subset@data[c('forest_spp_richness', 
  'tax_beta_div', 'func_beta_div', 'phyl_beta_div')], hist.col = NULL)
```
  
`r figRef("div_plots")` shows that none of the response variables was strongly correlated with species richness. All three beta diversity measures were positively correlated with each other. The correlation between functional and phylogenetic beta diversity was fairly strong.
  
##Taxonomic beta diversity  
```{r tax_beta_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('tax_beta_model_summary', 'Summary of linear model explaining taxonomic beta diversity'))
tax_beta_model_coef <- as.data.frame(tax_beta_model_summary$coefficients)
colnames(tax_beta_model_coef)[4] <- 'p value'
rownames(tax_beta_model_coef) <- expl_names
pander::pander(tax_beta_model_coef)

tax_beta_range <- round(max(units_merged_beta_subset$tax_beta_div) - 
    min(units_merged_beta_subset$tax_beta_div), digits = 3)

tax_beta_hdiv_change <- round(tax_beta_model_coef['Habitat diversity', 'Estimate'] * hdiv_range, digits = 2)
tax_beta_hdiv_perc <- round((tax_beta_hdiv_change / tax_beta_range) * 100, digits = 2)
```

Taxonomic beta diversity was not impacted by human footprint index (HFI), nor forest loss. Taxonomic beta diversity was positively associated with habitat diversity. Along the range of habitat diversity values (`r tabRef("explanatory_summary")`), taxonomic beta diversity increased by `r tax_beta_hdiv_change` units (`r tax_beta_hdiv_perc`%).
  
##Functional beta diversity  
```{r func_beta_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('func_beta_model_summary', 'Summary of generalised linear model explaining functional beta diversity'))
func_beta_model_coef <- as.data.frame(func_beta_model_summary$coefficients)
colnames(func_beta_model_coef)[4] <- 'p value'
rownames(func_beta_model_coef) <- expl_names
pander::pander(func_beta_model_coef)

func_beta_range <- round(max(units_merged_beta_subset$func_beta_div) - 
    min(units_merged_beta_subset$func_beta_div), digits = 2)

func_beta_hfp_change <- round(func_beta_model_coef['Human footprint', 'Estimate'] * hfp_range, digits = 2)
func_beta_hfp_perc <- round((func_beta_hfp_change / func_beta_range) * 100, digits = 2)

func_beta_npp_change <- round(func_beta_model_coef['Net primary production', 'Estimate'] * npp_range, digits = 2)
func_beta_npp_perc <- round((func_beta_npp_change / func_beta_range) * 100, digits = 2)

func_beta_temp_change <- round(func_beta_model_coef['Mean temperature', 'Estimate'] * temp_range, digits = 2)
func_beta_temp_perc <- round((func_beta_temp_change / func_beta_range) * 100, digits = 2)
```
  
Functional beta diversity had a statistically significant association with HFI. Along the gradient of HFI in our data (`r tabRef("explanatory_summary")`), functional beta diversity increased strongly (`r func_beta_hfp_change` units, `r func_beta_hfp_perc`%). Forest loss did not appear to influence functional beta diversity. Interestingly, net primary production (NPP) had a strong negative impact on functional beta diversity. Along the range of NPP values functional beta diversity decreased by `r abs(func_beta_npp_change)` units (`r func_beta_npp_perc`%). Similarly, mean annual temperature had a negative impact, decreasing functional beta diversity by `r abs(func_beta_temp_change)` units (`r func_beta_temp_perc`%).
  
##Phylogenetic beta diversity  
```{r phyl_beta_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('phyl_beta_model_summary', 'Summary of spatial error model explaining phylogenetic beta diversity'))
phyl_beta_model_coef <- as.data.frame(phyl_beta_model_summary$Coef)
colnames(phyl_beta_model_coef)[4] <- 'p value'
rownames(phyl_beta_model_coef) <- expl_names
pander::pander(phyl_beta_model_coef)

phyl_beta_range <- round(max(units_merged_beta_subset$phyl_beta_div) - 
    min(units_merged_beta_subset$phyl_beta_div), digits = 3)

phyl_beta_npp_perc <- 100 - round(exp(phyl_beta_model_coef['Net primary production', 'Estimate'] * npp_range) * 100, digits = 2)
phyl_beta_npp_change <- round((phyl_beta_npp_perc / 100) * phyl_beta_range, digits = 3)

phyl_beta_temp_perc <- 100 - round(exp(phyl_beta_model_coef['Mean temperature', 'Estimate'] * temp_range) * 100, digits = 2)
phyl_beta_temp_change <- round((phyl_beta_temp_perc / 100) * phyl_beta_range, digits = 3)
```
  
Similar to taxonomic beta diversity, phylogenetic beta diversity was not impacted by HFI, nor forest loss. Phylogenetic beta diversity was negatively associated with NPP and mean temperature, although the biological impacts much stronger. Along the range of NPP values (`r tabRef("explanatory_summary")`) phylogenetic beta diversity decreased by `r abs(phyl_beta_npp_change)` units (-`r phyl_beta_npp_perc`%). Along the range of mean annual temperature values phylogenetic beta diversity decreased by `r abs(phyl_beta_temp_change)` units (-`r phyl_beta_temp_perc`%).
<br>

#Discussion  

* Human influence did not clearly impact beta diversity in our study
* No impact of human influence on taxonomic and phylogenetic beta diversity
* Positive impact of human footprint index (HFI) on functional beta diversity

* Why did HFI have a positive impact on functional beta diversity?
* Habitat diversity was taken into account, so increased beta diversity due increased habitat diversity caused by increased human influence should not be the case
* Could HFI cause increases in habitat diversity at scales finer than the environmental data?
* Bird counts were conducted from road sides, so bird communities were not in pristine areas
* Increased beta diversity as a consequence of HFI could be due to decreased alpha or increased gamma, but neither was statistically significant. Based on the results, it is possible that HFI had a statistically non-significant negative impact on functional alpha diversity, which was significant in the model explaining functional beta diversity.

* Forest loss did not impact any of the beta diversity measures
* Forest loss had a consistently negative impact on functional and phylogenetic alpha and gamma diversities
* Since forest loss had a negative impact on both alpha and gamma diversities, the method of calculating beta diversity (gamma divided by alpha) canceled the impact of forest loss on beta diversity. 
  
* Beta-diversity indices were not positive functions of net primary productivity or ambient temperature.
* We found that beta-diversities were negatively related to NPP and mean temperature. **Why is this?**

* What do our results say about the impacts of human influence on biotic homogenisation?
* 

<br>

#Conclusion  
<br>

#Appdendix  
##Taxonomic alpha diversity results
```{r tax_alpha_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('tax_alpha_model_summary', 'Summary of spatial error model explaining taxonomic alpha diversity'))
tax_alpha_model_coef <- as.data.frame(tax_alpha_model_summary$Coef)
colnames(tax_alpha_model_coef)[4] <- 'p value'
rownames(tax_alpha_model_coef) <- expl_names
pander::pander(tax_alpha_model_coef)

tax_alpha_range <- round(max(units_merged_beta_subset$tax_alpha_div) - 
    min(units_merged_beta_subset$tax_alpha_div), digits = 3)
```

##Taxonomic gamma diversity results
```{r tax_gamma_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('tax_gamma_model_summary', 'Summary of spatial error model explaining taxonomic gamma diversity'))
tax_gamma_model_coef <- as.data.frame(tax_gamma_model_summary$Coef)
colnames(tax_gamma_model_coef)[4] <- 'p value'
rownames(tax_gamma_model_coef) <- expl_names
pander::pander(tax_gamma_model_coef)

tax_gamma_range <- round(max(units_merged_beta_subset$tax_gamma_div) - 
    min(units_merged_beta_subset$tax_gamma_div), digits = 3)

tax_gamma_hdiv_change <- round(tax_gamma_model_coef['Habitat diversity', 'Estimate'] * hdiv_range, digits = 2)
tax_gamma_hdiv_perc <- round((tax_gamma_hdiv_change / tax_gamma_range) * 100, digits = 2)
```

##Functional alpha diversity results
```{r func_alpha_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('func_alpha_model_summary', 'Summary of spatial error model explaining functional alpha diversity'))
func_alpha_model_coef <- as.data.frame(func_alpha_model_summary$Coef)
colnames(func_alpha_model_coef)[4] <- 'p value'
rownames(func_alpha_model_coef) <- expl_names
pander::pander(func_alpha_model_coef)

func_alpha_range <- round(max(units_merged_beta_subset$func_alpha_div) - 
    min(units_merged_beta_subset$func_alpha_div), digits = 3)

func_alpha_forest_change <- round(func_alpha_model_coef['Forest loss', 'Estimate'] * forest_range, digits = 2)
func_alpha_forest_perc <- round((func_alpha_forest_change / func_alpha_range) * 100, digits = 2)

func_alpha_npp_change <- round(func_alpha_model_coef['Net primary production', 'Estimate'] * npp_range, digits = 2)
func_alpha_npp_perc <- round((func_alpha_npp_change / func_alpha_range) * 100, digits = 2)

func_alpha_temp_change <- round(func_alpha_model_coef['Mean temperature', 'Estimate'] * temp_range, digits = 2)
func_alpha_temp_perc <- round((func_alpha_temp_change / func_alpha_range) * 100, digits = 2)

func_alpha_hfp_change <- round(func_alpha_model_coef['Human footprint', 'Estimate'] * hfp_range, digits = 2)
func_alpha_hfp_perc <- round((func_alpha_hfp_change / func_alpha_range) * 100, digits = 2)
```

##Functional gamma diversity results
```{r func_gamma_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('func_gamma_model_summary', 'Summary of spatial error model explaining functional gamma diversity'))
func_gamma_model_coef <- as.data.frame(func_gamma_model_summary$Coef)
colnames(func_gamma_model_coef)[4] <- 'p value'
rownames(func_gamma_model_coef) <- expl_names
pander::pander(func_gamma_model_coef)

func_gamma_range <- round(max(units_merged_beta_subset$func_gamma_div) - 
    min(units_merged_beta_subset$func_gamma_div), digits = 3)

func_gamma_forest_change <- round(func_gamma_model_coef['Forest loss', 'Estimate'] * forest_range, digits = 2)
func_gamma_forest_perc <- round((func_gamma_forest_change / func_gamma_range) * 100, digits = 2)

func_gamma_temp_change <- round(func_gamma_model_coef['Mean temperature', 'Estimate'] * temp_range, digits = 2)
func_gamma_temp_perc <- round((func_gamma_temp_change / func_gamma_range) * 100, digits = 2)

func_gamma_hfp_change <- round(func_gamma_model_coef['Human footprint', 'Estimate'] * hfp_range, digits = 4)
func_gamma_hfp_perc <- round((func_gamma_hfp_change / func_alpha_range) * 100, digits = 2)
```

##Phylogenetic alpha diversity results
```{r phyl_alpha_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('phyl_alpha_model_summary', 'Summary of spatial error model explaining phylogenetic alpha diversity'))
phyl_alpha_model_coef <- as.data.frame(phyl_alpha_model_summary$Coef)
colnames(phyl_alpha_model_coef)[4] <- 'p value'
rownames(phyl_alpha_model_coef) <- expl_names
pander::pander(phyl_alpha_model_coef)

phyl_alpha_range <- round(max(units_merged_beta_subset$phyl_alpha_div) - 
    min(units_merged_beta_subset$phyl_alpha_div), digits = 3)

phyl_alpha_hfp_change <- round(phyl_alpha_model_coef['Human footprint', 'Estimate'] * hfp_range, digits = 2)
phyl_alpha_hfp_perc <- round((phyl_alpha_hfp_change / phyl_alpha_range) * 100, digits = 2)

phyl_alpha_forest_change <- round(phyl_alpha_model_coef['Forest loss', 'Estimate'] * forest_range, digits = 2)
phyl_alpha_forest_perc <- round((phyl_alpha_forest_change / phyl_alpha_range) * 100, digits = 2)
```

##Phylogenetic gamma diversity results
```{r phyl_gamma_div, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRef('phyl_gamma_model_summary', 'Summary of spatial error model explaining phylogenetic gamma diversity'))
phyl_gamma_model_coef <- as.data.frame(phyl_gamma_model_summary$Coef)
colnames(phyl_gamma_model_coef)[4] <- 'p value'
rownames(phyl_gamma_model_coef) <- expl_names
pander::pander(phyl_gamma_model_coef)

phyl_gamma_range <- round(max(units_merged_beta_subset$phyl_gamma_div) - 
    min(units_merged_beta_subset$phyl_gamma_div), digits = 3)

phyl_gamma_forest_change <- round(phyl_gamma_model_coef['Forest loss', 'Estimate'] * forest_range, digits = 2)
phyl_gamma_forest_perc <- round((phyl_gamma_forest_change / phyl_gamma_range) * 100, digits = 2)
```

##1. Correlations between human footprint and human influence variables in Minnesota data
```{r hfp_plots, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('hfp_plots', "Distributions of human footprint and other variables describing human influence, and Spearman's rank correlations between variables")}
psych::pairs.panels(units_merged_human_impact_subset, hist.col = NULL, method = 'spearman')
```
  
##2. Correlations between explanatory variables and their distributions
```{r explanatory_plots, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('explanatory_plots', "Distributions of explanatory variables, and Spearman's rank correlations between variables")}
psych::pairs.panels(units_merged_beta_subset@data[c('human_footprint', 
  'forest_loss', 'habitat_div', 'net_prim_prod', 'mean_temp', 'forest_point_freq')], 
  hist.col = NULL, method = 'spearman')
```
  
##3. Variance inflation factors from models  
```{r vif, echo = FALSE, message = FALSE, warning = FALSE, results = 'asis'}
set.caption(tabRefApp('vif', 'Variance inflation factors of explanatory variables in models explaining taxonomic, functional, and phylogenetic beta diversity'))
vif_table <- as.data.frame(rbind(t(tax_beta_model_vif), t(func_beta_model_vif), t(phyl_beta_model_vif)))
rownames(vif_table) <- c('Taxonomic', 'Functional', 'Phylogenetic')
colnames(vif_table) <- expl_names[2:7]
pander::pander(vif_table)
```
  
##4. Histogram of number of forest BBA points per unit
```{r for_point_hist, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('for_point_hist', 'Histrogram of the number of forested BBA points per unit')}
ggplot(units_merged_beta_subset@data, aes(x = forest_point_freq)) + 
  geom_histogram(binwidth = 1,  fill = "grey", colour = "black") + 
  xlab('Number of forested points') + 
  ylab('Count') + 
  scale_x_continuous(limits = c(3, 20.1), expand = c(0, 0)) + 
  scale_y_continuous(limits = c(0, 40), expand = c(0, 0)) + 
  mytheme
```
  
##5. Spatial distribution of response variables and species richness
###a) Taxonomic beta diversity
```{r tax_div_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('tax_div_plot', 'Map showing the spatial distribution of taxonomic beta diversity in the state of Minnesota')}
ggplot(data = units_merged_beta_subset_fort, 
  aes_string(x = 'long', y = 'lat', fill = 'tax_beta_div', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Taxonomic \nbeta diversity') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
  
###b) Functional beta diversity
```{r func_div_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('func_div_plot', 'Map showing the spatial distribution of functional beta diversity in the state of Minnesota')}
ggplot(data = units_merged_beta_subset_fort, 
  aes_string(x = 'long', y = 'lat', fill = 'func_beta_div', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Functional \nbeta diversity') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
  
###c) Phylogenetic beta diversity
```{r phyl_div_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('phyl_div_plot', 'Map showing the spatial distribution of phylogenetic beta diversity in the state of Minnesota')}
ggplot(data = units_merged_beta_subset_fort, 
  aes_string(x = 'long', y = 'lat', fill = 'phyl_beta_div', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Phylogenetic \nbeta diversity') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
  
###d) Forest species richness
```{r spp_richness_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('spp_richness_plot', 'Map showing the spatial distribution of species richness in the state of Minnesota')}
ggplot(data = units_merged_beta_subset_fort, #units_merged_beta_subset_fort
  aes_string(x = 'long', y = 'lat', fill = 'forest_spp_richness', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Forest species \nrichness') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
  
##6. Spatial distribution of explanatory variables  
###a) Human footprint  
```{r hfp_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('hfp_plot', 'Map showing the spatial distribution of human footprint values in the state of Minnesota')}
ggplot(data = units_merged_fort, 
  aes_string(x = 'long', y = 'lat', fill = 'human_footprint', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Human \nfootprint') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
  
###b) Forest loss  
```{r for_loss_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('for_loss_plot', 'Map showing the spatial distribution of forest loss values in the state of Minnesota')}
ggplot(data = units_merged_fort, #units_merged_beta_subset_fort
  aes_string(x = 'long', y = 'lat', fill = 'forest_loss', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Forest \nloss') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
  
###c) Habitat diversity  
```{r hab_div_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('hab_div_plot', 'Map showing the spatial distribution of habitat diversity values in the state of Minnesota')}
ggplot(data = units_merged_fort, 
  aes_string(x = 'long', y = 'lat', fill = 'habitat_div', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Habitat \ndiversity') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
  
###d) Net primary production  
```{r npp_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('npp_plot', 'Map showing the spatial distribution of net primary productivity values in the state of Minnesota')}
ggplot(data = units_merged_fort, 
  aes_string(x = 'long', y = 'lat', fill = 'net_prim_prod', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Net primary \nproduction') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
   
###e) Mean temperature  
```{r mean_temp_plot, echo = FALSE, message = FALSE, warning = FALSE, fig.cap = figRefApp('mean_temp_plot', 'Map showing the spatial distribution of mean temperature values in the state of Minnesota')}
ggplot(data = units_merged_fort, 
  aes_string(x = 'long', y = 'lat', fill = 'mean_temp', group = 'group')) + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = 'lightgray', color = NA) + 
  geom_polygon(color = 'black', size = .05) + 
  scale_fill_gradientn(colours = rev(terrain.colors(10)), name = 'Mean \ntemperature') + 
  geom_polygon(data = minnesota, aes(long, lat, group = group), fill = NA, color = 'black') + 
  coord_equal() + theme_void()
```
  
##7. List of used R packages and their versions  
```{r list_packages, echo = FALSE, comment = NA}
for (package in sort(.packages())) {
  #print(paste(package, packageVersion(package), sep = ' '))
  message(package, ' ', packageVersion(package))
  #cat(paste(package, ' ', packageVersion(package), '\n'))
}
```